---
title: "The Travelling Particles"
author: "Jessica Song"
date: "24/08/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

```{r setup, results='hide', warning = F, message = F}
if (!requireNamespace('BiocManager', quietly = T)) install.packages('BiocManager') 
if (!require('knitr')) install.packages('knitr'); library('knitr')
if (!require('rmarkdown')) install.packages('rmarkdown'); library('rmarkdown')
if (!require('ggplot2')) install.packages('ggplot2'); library('ggplot2')
if (!require('FSA')) install.packages('FSA'); library('FSA')
if (!require('ggpubr')) install.packages('ggpubr'); library('ggpubr')
if (!require('vegan')) install.packages('vegan'); library('vegan')
if (!require('dplyr')) install.packages('dplyr'); library('dplyr')
if (!require('tibble')) install.packages('tibble'); library('tibble')
if (!require('stats')) install.packages('stats'); library('stats')
if (!require('RColorBrewer')) install.packages('RColorBrewer'); library('RColorBrewer')
if (!require('SRS')) install.packages('SRS'); library('SRS')
if (!require('otuSummary')) install.packages('otuSummary'); library('otuSummary')
if (!require('ape')) install.packages(('ape')); library('ape')
if (!require('picante')) install.packages(('picante'), dependencies = TRUE); library('picante')
if (!require('parallel')) install.packages(('parallel')); library('parallel')
if (!require('doSNOW')) install.packages(('doSNOW')); library('doSNOW')
if (!require('reshape2')) install.packages(('reshape2')); library('reshape2')
if (!require('UpSetR')) install.packages(('UpSetR')); library('UpSetR')
if (!require('ecodist')) install.packages(('ecodist')); library('ecodist')
if (!require('taxa')) install.packages(('taxa')); library('taxa')
if (!require('DESeq2')) BiocManager::install(("DESeq2")); library('DESeq2')
if (!require('betapart')) install.packages('betapart', type='source', dependencies = T, repos='https://cran.r-project.org'); library('betapart')
devtools::install_github('grunwaldlab/metacoder')
if (!require('metacoder')) install.package(('metacoder')); library('metacoder')
if (!require('phyloseq')) BiocManager::install(("phyloseq")); library('phyloseq')
if (!require('grDevices')) install.packages(('grDevices')); library('grDevices')



knitr::opts_chunk$set(cache=TRUE)


setwd("/Users/jsong/Desktop/R/Travelling_Particles")
```


### Import Data
The raw dataset in its entirety comprises 99 16S samples (P01 - P45 and W01 - W54) as well as 15 cDNA samples (A01 - G02) which were excluded from the scope of this study and were removed at a later step.

```{r importdata}
raw_ASV_table <- read.table("table-w-taxa.txt",
                            sep="\t", dec=".", header=TRUE, row.names = 1)
head(raw_ASV_table)
dim(raw_ASV_table) 

taxa_assign <- read.csv("taxa_assign.txt"
                 , sep="\t", dec=".", header=TRUE)
head(taxa_assign)
dim(taxa_assign)

tax_assign <- read.csv("tax_assign.txt",
                     sep = "\t", dec = ".", header = T)
head(tax_assign)

metadata <- read.table("metadata.txt",
                       sep="\t", dec=".", header=TRUE)
head(metadata)
dim(metadata) 
```

### Scaling by Ranked Subsampling (Normalization)

```{r normalization}
SRS_table <- SRS(as.data.frame(raw_ASV_table), 
                     Cmin = min(colSums(as.data.frame(raw_ASV_table))))
colSums(SRS_table)
head(SRS_table[1:20])

data_curve <- as.data.frame(raw_ASV_table)

sort(colSums(data_curve))

SRS_curve <- SRScurve(data_curve[16:ncol(data_curve)], step = 1000, sample = 4036, max.sample.size = 10000, 
         col = c(rep("#cc79a7", 45), rep("#0d98ba", 54)))

rownames(SRS_table) <- rownames(raw_ASV_table)
head(SRS_table)
dim(SRS_table)

SRS_table_dna <- as.data.frame(SRS_table[,-c(1:15)])
head(SRS_table_dna)
dim(SRS_table_dna)

SRS_dna_filtered <- as.data.frame(SRS_table_dna[apply(SRS_table_dna, 1, function(x) !all(x==0)),])
head(SRS_dna_filtered[1:20])
dim(SRS_dna_filtered)
```
### Combine dataset with taxonomy and metadata 

```{r combine}
SRS_dna_filtered_df <- as.data.frame(rownames_to_column(SRS_dna_filtered, var ="ASVs"))

SRS_dna_taxa <- left_join(SRS_dna_filtered_df, taxa_assign[-c(10:11)],  by = "ASVs")
SRS_dna_taxa <- tibble::column_to_rownames(SRS_dna_taxa, var = "ASVs")
head(SRS_dna_taxa)
dim(SRS_dna_taxa)


t_SRS_dna_filtered <- as.data.frame(t(SRS_dna_filtered))
t_SRS_dna_filtered <- rownames_to_column(t_SRS_dna_filtered, var = "sample_id")
SRS_dna_meta <- right_join(metadata, t_SRS_dna_filtered, by = "sample_id")
SRS_dna_meta <- tibble::column_to_rownames(SRS_dna_meta, var = "sample_id")
head(SRS_dna_meta)
dim(SRS_dna_meta)
```

### Subsetting by site and type

```{r subset}
# by site
####bremen
SRS_bremen <- subset(SRS_dna_meta, SRS_dna_meta$site == "a_bremen")
head(SRS_bremen[1:20])
dim(SRS_bremen)

SRS_bremen_df <- as.data.frame(t(SRS_bremen[12:ncol(SRS_bremen)]))
SRS_bremen_df <- SRS_bremen_df[apply(SRS_bremen_df, 1, function(x) !all(x==0)),]
head(SRS_bremen_df)
dim(SRS_bremen_df)

####brake
SRS_brake <- subset(SRS_dna_meta, SRS_dna_meta$site == "b_brake")
head(SRS_brake[1:20])
dim(SRS_brake)

SRS_brake_df <- as.data.frame(t(SRS_brake[12:ncol(SRS_brake)]))
SRS_brake_df <- SRS_brake_df[apply(SRS_brake_df, 1, function(x) !all(x==0)),]
head(SRS_brake_df)
dim(SRS_brake_df)

####bremerhaven
SRS_bremerhaven <- subset(SRS_dna_meta, SRS_dna_meta$site == "c_bremerhaven")
head(SRS_bremerhaven[1:20])
dim(SRS_bremerhaven)

SRS_bremerhaven_df <- as.data.frame(t(SRS_bremerhaven[12:ncol(SRS_bremerhaven)]))
SRS_bremerhaven_df <- SRS_bremerhaven_df[apply(SRS_bremerhaven_df, 1, function(x) !all(x==0)),]
head(SRS_bremerhaven_df)
dim(SRS_bremerhaven_df)

####helgoland
SRS_helgoland <- subset(SRS_dna_meta, SRS_dna_meta$site == "d_helgoland")
head(SRS_helgoland[1:20])
dim(SRS_helgoland)

SRS_helgoland_df <- as.data.frame(t(SRS_helgoland[12:ncol(SRS_helgoland)]))
SRS_helgoland_df <- SRS_helgoland_df[apply(SRS_helgoland_df, 1, function(x) !all(x==0)),]
head(SRS_helgoland_df)
dim(SRS_helgoland_df)

# by type
####hdpe
SRS_hdpe <- subset(SRS_dna_meta, SRS_dna_meta$type == "HDPE")
head(SRS_hdpe[1:20])
dim(SRS_hdpe)

SRS_hdpe_df <- as.data.frame(t(SRS_hdpe[12:ncol(SRS_hdpe)]))
SRS_hdpe_df <- SRS_hdpe_df[apply(SRS_hdpe_df, 1, function(x) !all(x==0)),]
head(SRS_hdpe_df)
dim(SRS_hdpe_df)

####tw
SRS_tw <- subset(SRS_dna_meta, SRS_dna_meta$type == "tyre_wear")
head(SRS_tw[1:20])
dim(SRS_tw)

SRS_tw_df <- as.data.frame(t(SRS_tw[12:ncol(SRS_tw)]))
SRS_tw_df <- SRS_tw_df[apply(SRS_tw_df, 1, function(x) !all(x==0)),]
head(SRS_tw_df)
dim(SRS_tw_df)

####wood
SRS_wood <- subset(SRS_dna_meta, SRS_dna_meta$type == "wood")
head(SRS_wood[1:20])
dim(SRS_wood)

SRS_wood_df <- as.data.frame(t(SRS_wood[12:ncol(SRS_wood)]))
SRS_wood_df <- SRS_wood_df[apply(SRS_wood_df, 1, function(x) !all(x==0)),]
head(SRS_wood_df)
dim(SRS_wood_df)

####water 3
SRS_water3 <- subset(SRS_dna_meta, SRS_dna_meta$type == "water_3")
head(SRS_water3[1:20])
dim(SRS_water3)

SRS_water3_df <- as.data.frame(t(SRS_water3[12:ncol(SRS_water3)]))
SRS_water3_df <- SRS_water3_df[apply(SRS_water3_df, 1, function(x) !all(x==0)),]
head(SRS_water3_df)
dim(SRS_water3_df)

####water 02
SRS_water02 <- subset(SRS_dna_meta, SRS_dna_meta$type == "water_0_2")
head(SRS_water02[1:20])
dim(SRS_water02)

SRS_water02_df <- as.data.frame(t(SRS_water02[12:ncol(SRS_water02)]))
SRS_water02_df <- SRS_water02_df[apply(SRS_water02_df, 1, function(x) !all(x==0)),]
head(SRS_water02_df)
dim(SRS_water02_df)
```

### Community composition
#### Bind subsets to taxonomy
```{r taxa}
# by site
####bremen
bremen_taxa <- as.data.frame(rownames_to_column(SRS_bremen_df, var ="ASVs"))
bremen_taxa <- left_join(bremen_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
bremen_taxa <- tibble::column_to_rownames(bremen_taxa, var = "ASVs")
head(bremen_taxa)
dim(bremen_taxa)

####brake
brake_taxa <- as.data.frame(rownames_to_column(SRS_brake_df, var ="ASVs"))
brake_taxa <- left_join(brake_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
brake_taxa <- tibble::column_to_rownames(brake_taxa, var = "ASVs")
head(brake_taxa)
dim(brake_taxa)

####bremerhaven
bremerhaven_taxa <- as.data.frame(rownames_to_column(SRS_bremerhaven_df, var ="ASVs"))
bremerhaven_taxa <- left_join(bremerhaven_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
bremerhaven_taxa <- tibble::column_to_rownames(bremerhaven_taxa, var = "ASVs")
head(bremerhaven_taxa)
dim(bremerhaven_taxa)

####helgoland
helgoland_taxa <- as.data.frame(rownames_to_column(SRS_helgoland_df, var ="ASVs"))
helgoland_taxa <- left_join(helgoland_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
helgoland_taxa <- tibble::column_to_rownames(helgoland_taxa, var = "ASVs")
head(helgoland_taxa)
dim(helgoland_taxa)

# by type
####hdpe
hdpe_taxa <- as.data.frame(rownames_to_column(SRS_hdpe_df, var ="ASVs"))
hdpe_taxa <- left_join(hdpe_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
hdpe_taxa <- tibble::column_to_rownames(hdpe_taxa, var = "ASVs")
head(hdpe_taxa)
dim(hdpe_taxa)

####tyre wear
tw_taxa <- as.data.frame(rownames_to_column(SRS_tw_df, var ="ASVs"))
tw_taxa <- left_join(tw_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
tw_taxa <- tibble::column_to_rownames(tw_taxa, var = "ASVs")
head(tw_taxa)
dim(tw_taxa)

####wood
wood_taxa <- as.data.frame(rownames_to_column(SRS_wood_df, var ="ASVs"))
wood_taxa <- left_join(wood_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
wood_taxa <- tibble::column_to_rownames(wood_taxa, var = "ASVs")
head(wood_taxa)
dim(wood_taxa)

####water 3
water3_taxa <- as.data.frame(rownames_to_column(SRS_water3_df, var ="ASVs"))
water3_taxa <- left_join(water3_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
water3_taxa <- tibble::column_to_rownames(water3_taxa, var = "ASVs")
head(water3_taxa)
dim(water3_taxa)

####water 02
water02_taxa <- as.data.frame(rownames_to_column(SRS_water02_df, var ="ASVs"))
water02_taxa <- left_join(water02_taxa, taxa_assign[-c(10:11)],  by = "ASVs")
water02_taxa <- tibble::column_to_rownames(water02_taxa, var = "ASVs")
head(water02_taxa)
dim(water02_taxa)
```

#### Taxa bar plots

```{r taxlevel}
# collapse at Class level
SRS_dna_class <- cbind(SRS_dna_taxa$class, SRS_dna_taxa[-c(100:ncol(SRS_dna_taxa))])
colnames(SRS_dna_class) <- c("class", colnames(SRS_dna_taxa[-c(100:ncol(SRS_dna_taxa))]))
SRS_dna_class

SRS_dna_class_collap <- otuCollap(SRS_dna_class, taxto = 1, taxhead = "class",
                                    pattern = "_____________")
colnames(SRS_dna_class_collap) <- colnames(SRS_dna_class)
SRS_dna_class_collap
dim(SRS_dna_class_collap)

## relative abundances of classes were calculated for each type and imported 
dna_class_relabund <- read.table("dna_class_relabund.txt",
                                 sep="\t", dec=".", header=TRUE)
dna_class_relabund
dim(dna_class_relabund)


dna_class_relabund$class <- ifelse(rowSums(dna_class_relabund[2:26]/ncol(dna_class_relabund[2:26])) < 1, "<1 % Class", as.character(dna_class_relabund$class)) #renaming all class under 1%
rare_class <- dna_class_relabund[which(dna_class_relabund$class == "<1 % Class"),] #show list of class <0.1%
rare_class <- colSums(rare_class[2:26]) #calculating total relative abundance of new "Rare" group per sample 
dna_class_relabund <- dna_class_relabund[-which(dna_class_relabund$class == "<1 % Class"),]
rare_class <- append(rare_class, "<1 % Class (102 other taxa)") #adding col names
names(rare_class) <- c("bremen.HDPE", "bremen.wood", "bremen.tw", "brake.HDPE", "brake.wood", "brake.tw", 
                       "bremerhaven.HDPE", "bremerhaven.wood", "bremerhaven.tw", "helgolandS.HDPE", "helgolandS.wood", "helgolandS.tw", 
                       "helgolandT.HDPE", "helgolandT.wood", "helgolandT.tw", "bremen.water3", "bremen.water02", "helgolandS.water3", "helgolandS.water02", 
                       "brake.water3", "brake.water02", "bremerhaven.water3", "bremerhaven.water02", "helgolandT.water3", "helgolandT.water02", "class")
rare_class <- rare_class[c(26,1:25)] #bring class name to first row
dna_class_relabund <- rbind(dna_class_relabund,as.data.frame(t(rare_class)))
dim(dna_class_relabund)
head(dna_class_relabund)

#make all numeric
for (i in 2:length(dna_class_relabund)) {
  dna_class_relabund[,i] <- as.numeric(dna_class_relabund[,i])
}

dna_class_relabund
dim(dna_class_relabund)

#stretching palette
n_class <- nrow(dna_class_relabund)
palette_class <- colorRampPalette(RColorBrewer::brewer.pal(11, "BrBG"))(n_class)


# subsetting by type
####hdpe
class_hdpe <- dna_class_relabund[c("class", "bremen.HDPE", "brake.HDPE", "bremerhaven.HDPE", "helgolandT.HDPE", "helgolandS.HDPE")]
colnames(class_hdpe) <- c("class", "Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")
class_hdpe
dim(class_hdpe)

#change to long format
class_hdpe_melt <- melt(class_hdpe)
class_hdpe_melt$class <- factor(class_hdpe_melt$class, levels = class_hdpe$class) 

#define variables
class_hdpe_melt$sites <- plyr::mapvalues(class_hdpe_melt$variable, 
                                   c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"), 
                                   c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"))
class_hdpe_melt

####tyre wear
class_tw <- dna_class_relabund[c("class", "bremen.tw", "brake.tw", "bremerhaven.tw", "helgolandT.tw", "helgolandS.tw")]
colnames(class_tw) <- c("class", "Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")
class_tw
dim(class_tw)

#change to long format
class_tw_melt <- melt(class_tw)
class_tw_melt$class <- factor(class_tw_melt$class, levels = class_tw$class) 

#define variables
class_tw_melt$sites <- plyr::mapvalues(class_tw_melt$variable, 
                                 c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"), 
                                 c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"))
class_tw_melt

####wood
class_wood <- dna_class_relabund[c("class", "bremen.wood", "brake.wood", "bremerhaven.wood", "helgolandT.wood", "helgolandS.wood")]
colnames(class_wood) <- c("class", "Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")
class_wood
dim(class_wood)

#change to long format
class_wood_melt <- melt(class_wood)
class_wood_melt$class <- factor(class_wood_melt$class, levels = class_wood$class) 

#define variables
class_wood_melt$sites <- plyr::mapvalues(class_wood_melt$variable, 
                                   c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"), 
                                   c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"))
class_wood_melt

####water 3
class_water3 <- dna_class_relabund[c("class", "bremen.water3", "brake.water3", "bremerhaven.water3", "helgolandT.water3", "helgolandS.water3")]
colnames(class_water3) <- c("class", "Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")
class_water3
dim(class_water3)

#change to long format
class_water3_melt <- melt(class_water3)
class_water3_melt$class <- factor(class_water3_melt$class, levels = class_water3$class) 

#define variables
class_water3_melt$sites <- plyr::mapvalues(class_water3_melt$variable, 
                                     c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"), 
                                     c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"))
class_water3_melt

####water 02
class_water02 <- dna_class_relabund[c("class", "bremen.water02", "brake.water02", "bremerhaven.water02", "helgolandT.water02", "helgolandS.water02")]
colnames(class_water02) <- c("class", "Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")
class_water02
dim(class_water02)

#change to long format
class_water02_melt <- melt(class_water02)
class_water02_melt$class <- factor(class_water02_melt$class, levels = class_water02$class) 

#define variables
class_water02_melt$sites <- plyr::mapvalues(class_water02_melt$variable, 
                                      c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"), 
                                      c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)"))
class_water02_melt


# visualization: class bar plots
bar_theme <- theme(axis.text.y = element_text(colour = "black", size = 16),
        axis.text.x = element_text(colour = "black", size = 16, angle = 90, vjust = 0.5, hjust = 1),
        legend.text = element_text(colour = "black", size = 16),
        legend.position = "right",
        legend.key = element_blank(),
        axis.ticks = element_line(size = 0.75),
        axis.ticks.length = unit(0.2, "cm"),
        axis.line = element_line(colour = "black", size = 0.75),
        legend.title = element_text(colour = "black", face = "bold", size = 16),
        panel.background = element_blank(), 
        panel.border = element_blank(),
        plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 20))


(class_hdpe_plot <- ggplot(class_hdpe_melt, aes(x = sites, y = value)) + 
  geom_bar(aes(fill = class), color = "#FFFFFF", position = position_stack(), stat = "identity") +
  theme(plot.margin = margin(t = 1, r = -2, b = 1, l = 1, unit = "cm"),
        axis.title.y = element_text(colour = "black", size = 18, vjust = 3),
        axis.title.x = element_blank()) +
  coord_fixed(ratio = 0.05) +
  scale_fill_manual(values = palette_class) +
  labs(x = "Sampling Site", y = "Relative Abundance [%]") +
  ggtitle("HDPE") +
  bar_theme)

(class_tw_plot <- ggplot(class_tw_melt, aes(x = sites, y = value)) + 
  geom_bar(aes(fill = class), color = "#FFFFFF", position = position_stack(), stat = "identity") +
  theme(plot.margin = margin(t = 1, r = -2, b = 1, l = -2, unit = "cm"),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) +
  coord_fixed(ratio = 0.05) +
  scale_fill_manual(values = palette_class) +
  labs(x = "Sampling Site", y = "Relative Abundance [%]") +
  ggtitle("Tyre Wear") +
  bar_theme)

(class_wood_plot <- ggplot(class_wood_melt, aes(x = sites, y = value)) + 
  geom_bar(aes(fill = class), color = "#FFFFFF", position = position_stack(), stat = "identity") +
  theme(plot.margin = margin(t = 1, r = -2, b = 1, l = -2, unit = "cm"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(colour = "black", size = 18, vjust = -5)) +
  coord_fixed(ratio = 0.05) +
  scale_fill_manual(values = palette_class) +
  labs(x = "Sampling Site", y = "Relative Abundance [%]") +
  ggtitle("Wood") +
  bar_theme)

(class_water3_plot <- ggplot(class_water3_melt, aes(x = sites, y = value)) + 
  geom_bar(aes(fill = class), color = "#FFFFFF", position = position_stack(), stat = "identity") +
  theme(plot.margin = margin(t = 1, r = -2, b = 1, l = -2, unit = "cm"),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) +
  coord_fixed(ratio = 0.05) +
  scale_fill_manual(values = palette_class) +
  labs(x = "Sampling Site", y = "Relative Abundance [%]") +
  ggtitle("Water 3") +
  bar_theme)

(class_water02_plot <- ggplot(class_water02_melt, aes(x = sites, y = value)) + 
  geom_bar(aes(fill = class), color = "#FFFFFF", position = position_stack(), stat = "identity") +
  theme(plot.margin = margin(t = 1, r = 1, b = 1, l = -2, unit = "cm"),
        axis.title.y = element_blank(),
        axis.title.x = element_blank()) +
  coord_fixed(ratio = 0.05) +
  scale_fill_manual(values = palette_class) +
  labs(x = "Sampling Site", y = "Relative Abundance [%]") +
  ggtitle("Water 02") +
  bar_theme)

# combine plots
(class_bar <- ggarrange(class_hdpe_plot, class_tw_plot, class_wood_plot, class_water3_plot, class_water02_plot, 
                       common.legend = T, legend = "right", nrow = 1, ncol = 5, align = "hv"))

ggsave("class_bar_plot.tiff", class_bar, width = 500, height = 285, units = "mm", dpi = 300)
```

#### Differential abundance testing

```{r diffabund}
# preparing raw data for testing
raw_ASV <- as.data.frame(raw_ASV_table[16:ncol(raw_ASV_table)])
dim(raw_ASV)
raw_ASV <- rownames_to_column(raw_ASV, var = "ASVs")

ASV_dna_tax <- left_join(raw_ASV, taxa_assign, by = "ASVs")
head(ASV_dna_tax)
dim(ASV_dna_tax)

# renaming taxonomy
ASV_dna_tax$taxonomy <- sub(ASV_dna_tax$taxonomy,
                             pattern = "D_0", replacement = "k")
ASV_dna_tax$taxonomy <- sub(ASV_dna_tax$taxonomy,
                             pattern = "D_1", replacement = "p")
ASV_dna_tax$taxonomy <- sub(ASV_dna_tax$taxonomy,
                             pattern = "D_2", replacement = "c")
ASV_dna_tax$taxonomy <- sub(ASV_dna_tax$taxonomy,
                             pattern = "D_3", replacement = "o")
ASV_dna_tax$taxonomy <- sub(ASV_dna_tax$taxonomy,
                             pattern = "D_4", replacement = "f")
ASV_dna_tax$taxonomy <- sub(ASV_dna_tax$taxonomy,
                             pattern = "D_5", replacement = "g")
ASV_dna_tax$taxonomy <- sub(ASV_dna_tax$taxonomy,
                             pattern = "D_6", replacement = "s")
ASV_dna_tax


#parsing taxonomies
ASV_dna_taxmap <- parse_tax_data(ASV_dna_tax, class_cols = "taxonomy", class_sep = ";",
                                  class_regex = "^([a-z]{0,1})_{0,2}(.*)$", class_key = c(tax_rank = "taxon_rank", tax_name = "taxon_name"))

ASV_dna_taxmap

#abundance per-taxon
ASV_dna_taxmap$data$tax_abund <- calc_taxon_abund(ASV_dna_taxmap, "tax_data", cols = metadata$sample_id)
ASV_dna_taxmap


#number of samples that have reads for each taxon (taxon occupancy/occurrence)
ASV_dna_taxmap$data$tax_occ <- calc_n_samples(ASV_dna_taxmap, "tax_abund", groups = metadata$type, 
                                               cols = metadata$sample_id)
ASV_dna_taxmap

#comparing between sites using DESeq2
ASV_dna_taxmap$data$diff_table <- calc_diff_abund_deseq2(ASV_dna_taxmap, data = "tax_abund",
                                                          cols = metadata$sample_id,
                                                          groups = metadata$type)
range(ASV_dna_taxmap$data$diff_table$padj, finite = TRUE)


set.seed(1)

#tree subset to order rank
ASV_dna_taxmap %>% 
  metacoder::filter_taxa(taxon_ranks == "o", supertaxa = T, reassign_obs = F) %>%
  heat_tree_matrix(data = "diff_table",
                   node_label = taxon_names,
                   node_size = n_obs,
                   node_color = ifelse(is.na(padj) | padj > 0.05, 0, log2FoldChange),
                   node_color_range = diverging_palette(),  # The built-in palette for diverging data
                   node_color_trans = "linear", # The default is scaled by circle area
                   node_color_interval = c(-3, 3), # The range of `log2_fold_change` to display
                   edge_color_interval = c(-3, 3), # The range of `log2_fold_change` to display
                   node_size_axis_label = "Number of ASVs",
                   node_color_axis_label = "Log2 fold change",
                   layout = "davidson-harel", # The primary layout algorithm
                   initial_layout = "reingold-tilford", # The layout algorithm that initializes node locations
                   output_file = "differential_heat_tree_dna_raw_o.pdf", # Saves the plot as a pdf file
                   verbose = T)
```

### Taxonomic alpha diversity

```{r taxalpha}
# Local taxonomic diversity
shannon_div <- vegan::diversity(SRS_dna_meta[12:ncol(SRS_dna_meta)], index = "shannon", MARGIN = 1) # shannon index
shannon_div
summary(shannon_div)


simpsons <- vegan::diversity(SRS_dna_meta[12:ncol(SRS_dna_meta)], index = "simpson", MARGIN = 1) # simpson's index
simpsons
summary(simpsons)


data_scores_alpha <- as.data.frame(cbind(scores(shannon_div), scores(simpsons)))
colnames(data_scores_alpha) <- c("shannon", "simpson")
data_scores_alpha$type <- SRS_dna_meta$type
data_scores_alpha$site_x_cage <- SRS_dna_meta$site_x_cage

data_scores_alpha

data_scores_alpha$type <- factor(data_scores_alpha$type, 
                                      levels = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"))
data_scores_alpha$site_x_cage <- factor(data_scores_alpha$site_x_cage, 
                                 levels = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"))

# Rank sums and post hoc test
kruskal.test(shannon_div~SRS_dna_meta$site_x_cage) # shannon by site
shan.dunn.site <- dunnTest(shannon_div, as.factor(SRS_dna_meta$site_x_cage), method="bh", list = T) 

shan_dunn_site <- as.data.frame(shan.dunn.site$res)
shan_dunn_site <- tibble::column_to_rownames(shan_dunn_site, var = "Comparison")
shan_dunn_site


kruskal.test(shannon_div~SRS_dna_meta$cage_x_type) # shannon by type
shan.dunn.type <- dunnTest(shannon_div, as.factor(SRS_dna_meta$cage_x_type), method="bh") 

shan_dunn_type <- as.data.frame(shan.dunn.type$res)
shan_dunn_type <- tibble::column_to_rownames(shan_dunn_type, var = "Comparison")
shan_dunn_type


kruskal.test(simpsons~SRS_dna_meta$site_x_cage) # simpson by site
simp.dunn.site <- dunnTest(simpsons, as.factor(SRS_dna_meta$site_x_cage), method="bh") 

simp_dunn_site <- as.data.frame(simp.dunn.site$res)
simp_dunn_site <- tibble::column_to_rownames(simp_dunn_site, var = "Comparison")
simp_dunn_site

kruskal.test(simpsons~SRS_dna_meta$cage_x_type) # simpson by type
simp.dunn.type <- dunnTest(simpsons, as.factor(SRS_dna_meta$cage_x_type), method="bh") 

simp_dunn_type <- as.data.frame(simp.dunn.type$res)
simp_dunn_type <- tibble::column_to_rownames(simp_dunn_type, var = "Comparison")
simp_dunn_type
```

### Taxonomic beta diversity

```{r taxbeta}
# Taxonomic dissimilarities between samples
SRS_dna_meta_df <- as.data.frame(SRS_dna_meta[14:ncol(SRS_dna_meta)])
head(SRS_dna_meta_df[1:20])

transformation <- sqrt # square root transformation

dna_BC <- transformation(SRS_dna_meta_df)
dna_BC_matrix <- as.matrix(vegdist(dna_BC, method = "bray"))
head(dna_BC_matrix)

dna_JC <- transformation(SRS_dna_meta_df)
dna_JC_matrix <- as.matrix(vegdist(dna_JC, method = "jaccard", binary = TRUE))
head(dna_JC_matrix)

# Taxonomic dissimilarities between types at each site (weighted)
## Bray Curtis
#### bremen
bremen <- as.data.frame(t(SRS_bremen_df))
bremen_BC <- transformation(bremen)
bremen_BC_matrix <- as.matrix(vegdist(bremen_BC, method = "bray"))
head(bremen_BC_matrix)

#### brake
brake <- as.data.frame(t(SRS_brake_df))
brake_BC <- transformation(brake)
brake_BC_matrix <- as.matrix(vegdist(brake_BC, method = "bray"))
head(brake_BC_matrix)

####bremerhaven
bremerhaven <- as.data.frame(t(SRS_bremerhaven_df))
bremerhaven_BC <- transformation(bremerhaven)
bremerhaven_BC_matrix <- as.matrix(vegdist(bremerhaven_BC, method = "bray"))
head(bremerhaven_BC_matrix)

####helgoland
helgoland <- as.data.frame(t(SRS_helgoland_df))
helgoland_BC <- transformation(helgoland)
helgoland_BC_matrix <- as.matrix(vegdist(helgoland_BC, method = "bray"))
head(helgoland_BC_matrix)

# Taxonomic dissimilarities within each type across sites (weighted)
## Bray Curtis
####hdpe
hdpe <- as.data.frame(t(SRS_hdpe_df))
hdpe_BC <- transformation(hdpe)
hdpe_BC_matrix <- as.matrix(vegdist(hdpe_BC, method = "bray"))
head(hdpe_BC_matrix)

####tyre wear
tw <- as.data.frame(t(SRS_tw_df))
tw_BC <- transformation(tw)
tw_BC_matrix <- as.matrix(vegdist(tw_BC, method = "bray"))
head(tw_BC_matrix)

####wood
wood <- as.data.frame(t(SRS_wood_df))
wood_BC <- transformation(wood)
wood_BC_matrix <- as.matrix(vegdist(wood_BC, method = "bray"))
head(wood_BC_matrix)

####water 3
water3 <- as.data.frame(t(SRS_water3_df))
water3_BC <- transformation(water3)
water3_BC_matrix <- as.matrix(vegdist(water3_BC, method = "bray"))
head(water3_BC_matrix)

####water 02
water02 <- as.data.frame(t(SRS_water02_df))
water02_BC <- transformation(water02)
water02_BC_matrix <- as.matrix(vegdist(water02_BC, method = "bray"))
head(water02_BC_matrix)

# Taxonomic dissimilarities between types at each site (unweighted)
## Jaccard
#### bremen
bremen_JC_matrix <- as.matrix(vegdist(bremen, method = "jaccard", binary = TRUE))
head(bremen_JC_matrix)

#### brake
brake_JC_matrix <- as.matrix(vegdist(brake, method = "jaccard", binary = TRUE))
head(brake_JC_matrix)

####bremerhaven
bremerhaven_JC_matrix <- as.matrix(vegdist(bremerhaven, method = "jaccard", binary = TRUE))
head(bremerhaven_JC_matrix)

####helgoland
helgoland_JC_matrix <- as.matrix(vegdist(helgoland, method = "jaccard", binary = TRUE))
head(helgoland_JC_matrix)

# Taxonomic dissimilarities within each type across sites (unweighted)
## Jaccard
####hdpe
hdpe_JC_matrix <- as.matrix(vegdist(hdpe, method = "jaccard", binary = TRUE))
head(hdpe_JC_matrix)

####tyre wear
tw_JC_matrix <- as.matrix(vegdist(tw, method = "jaccard", binary = TRUE))
head(tw_JC_matrix)

####wood
wood_JC_matrix <- as.matrix(vegdist(wood, method = "jaccard", binary = TRUE))
head(wood_JC_matrix)

####water 3
water3_JC_matrix <- as.matrix(vegdist(water3, method = "jaccard", binary = TRUE))
head(water3_JC_matrix)

####water 02
water02_JC_matrix <- as.matrix(vegdist(water02, method = "jaccard", binary = TRUE))
head(water02_JC_matrix)
```

### Preparing for phylogenetic analyses

```{r phy}
# Phylogenetic tree is linked to original OTU IDs so these are restored here in place of the ASV numbers assigned
phy_SRS_dna <- SRS_dna_filtered_df[,-1]
rownames(phy_SRS_dna) <- SRS_dna_taxa$OTU_ID

taxa <- as.data.frame(taxa_assign[c(1:8)])
rownames(taxa) <- taxa_assign[,9]
phy_taxa <- taxa[-c(1)]
head(phy_taxa)

phy_meta <- column_to_rownames(metadata, var = "sample_id")


# create phyloseq object
asvs <- otu_table(as.matrix(phy_SRS_dna), taxa_are_rows = T) 
tax <- tax_table(as.matrix(phy_taxa))
meta <- sample_data(phy_meta)

taxa_names(asvs) # check if taxa names match
taxa_names(tax)

physeq <- phyloseq(asvs, tax, meta)

tree <- read.tree("tree.nwk")

physeq_tree <- merge_phyloseq(physeq, tree)

# subsetting by site
####bremen
physeq_bremen <- subset_samples(physeq_tree, site=="a_bremen")
physeq_bremen <- prune_taxa(taxa_sums(physeq_bremen) > 0, physeq_bremen)
physeq_bremen

####brake
physeq_brake <- subset_samples(physeq_tree, site=="b_brake")
physeq_brake <- prune_taxa(taxa_sums(physeq_brake) > 0, physeq_brake)
physeq_brake

####bremerhaven
physeq_bremerhaven <- subset_samples(physeq_tree, site=="c_bremerhaven")
physeq_bremerhaven <- prune_taxa(taxa_sums(physeq_bremerhaven) > 0, physeq_bremerhaven)
physeq_bremerhaven

####helgoland
physeq_helgoland <- subset_samples(physeq_tree, site=="d_helgoland")
physeq_helgoland <- prune_taxa(taxa_sums(physeq_helgoland) > 0, physeq_helgoland)
physeq_helgoland

# subsetting by type
####hdpe
physeq_hdpe <- subset_samples(physeq_tree, type=="HDPE")
physeq_hdpe <- prune_taxa(taxa_sums(physeq_hdpe) > 0, physeq_hdpe)
physeq_hdpe

####tyre wear
physeq_tw <- subset_samples(physeq_tree, type=="tyre_wear")
physeq_tw <- prune_taxa(taxa_sums(physeq_tw) > 0, physeq_tw)
physeq_tw

####wood
physeq_wood <- subset_samples(physeq_tree, type=="wood")
physeq_wood <- prune_taxa(taxa_sums(physeq_wood) > 0, physeq_wood)
physeq_wood

####water3
physeq_water3 <- subset_samples(physeq_tree, type=="water_3")
physeq_water3 <- prune_taxa(taxa_sums(physeq_water3) > 0, physeq_water3)
physeq_water3

####water02
physeq_water02 <- subset_samples(physeq_tree, type=="water_0_2")
physeq_water02 <- prune_taxa(taxa_sums(physeq_water02) > 0, physeq_water02)
physeq_water02


## removing sequences from tree that are not present in ASV table
t_phy_SRS_dna <- t(phy_SRS_dna)
colSums(t_phy_SRS_dna)

pruned_tree <- prune.sample(t_phy_SRS_dna, tree)
pruned_tree
str(pruned_tree)
pruned_tree$tip.label

match_asv_phylo <- match.phylo.data(pruned_tree, phy_SRS_dna)
match_asv_phylo

# pruning tree by site for downstream analyses
####bremen
phy_bremen <- SRS_bremen_df
rownames(phy_bremen) <- bremen_taxa$OTU_ID
t_phy_bremen <- t(phy_bremen)

bremen_tree <- prune.sample(t_phy_bremen, tree)
str(bremen_tree)
bremen_tree$tip.label

match_asv_phylo_bremen <- match.phylo.data(bremen_tree, phy_bremen)
match_asv_phylo_bremen

####brake
phy_brake <- SRS_brake_df
rownames(phy_brake) <- brake_taxa$OTU_ID
t_phy_brake <- t(phy_brake)

brake_tree <- prune.sample(t_phy_brake, tree)
str(brake_tree)
brake_tree$tip.label

match_asv_phylo_brake <- match.phylo.data(brake_tree, phy_brake)
match_asv_phylo_brake

####bremerhaven
phy_bremerhaven <- SRS_bremerhaven_df
rownames(phy_bremerhaven) <- bremerhaven_taxa$OTU_ID
t_phy_bremerhaven <- t(phy_bremerhaven)

bremerhaven_tree <- prune.sample(t_phy_bremerhaven, tree)
str(bremerhaven_tree)
bremerhaven_tree$tip.label

match_asv_phylo_bremerhaven <- match.phylo.data(bremerhaven_tree, phy_bremerhaven)
match_asv_phylo_bremerhaven

####helgoland
phy_helgoland <- SRS_helgoland_df
rownames(phy_helgoland) <- helgoland_taxa$OTU_ID
t_phy_helgoland <- t(phy_helgoland)

helgoland_tree <- prune.sample(t_phy_helgoland, tree)
str(helgoland_tree)
helgoland_tree$tip.label

match_asv_phylo_helgoland <- match.phylo.data(helgoland_tree, phy_helgoland)
match_asv_phylo_helgoland

# pruning tree by type for downstream analyses
####hdpe
phy_hdpe <- SRS_hdpe_df
rownames(phy_hdpe) <- hdpe_taxa$OTU_ID
t_phy_hdpe <- t(phy_hdpe)

hdpe_tree <- prune.sample(t_phy_hdpe, tree)
str(hdpe_tree)
hdpe_tree$tip.label

match_asv_phylo_hdpe <- match.phylo.data(hdpe_tree, phy_hdpe)
match_asv_phylo_hdpe

####tw
phy_tw <- SRS_tw_df
rownames(phy_tw) <- tw_taxa$OTU_ID
t_phy_tw <- t(phy_tw)

tw_tree <- prune.sample(t_phy_tw, tree)
str(tw_tree)
tw_tree$tip.label

match_asv_phylo_tw <- match.phylo.data(tw_tree, phy_tw)
match_asv_phylo_tw

####wood
phy_wood <- SRS_wood_df
rownames(phy_wood) <- wood_taxa$OTU_ID
t_phy_wood <- t(phy_wood)

wood_tree <- prune.sample(t_phy_wood, tree)
str(wood_tree)
wood_tree$tip.label

match_asv_phylo_wood <- match.phylo.data(wood_tree, phy_wood)
match_asv_phylo_wood

####water3
phy_water3 <- SRS_water3_df
rownames(phy_water3) <- water3_taxa$OTU_ID
t_phy_water3 <- t(phy_water3)

water3_tree <- prune.sample(t_phy_water3, tree)
str(water3_tree)
water3_tree$tip.label

match_asv_phylo_water3 <- match.phylo.data(water3_tree, phy_water3)
match_asv_phylo_water3

####water02
phy_water02 <- SRS_water02_df
rownames(phy_water02) <- water02_taxa$OTU_ID
t_phy_water02 <- t(phy_water02)

water02_tree <- prune.sample(t_phy_water02, tree)
str(water02_tree)
water02_tree$tip.label

match_asv_phylo_water02 <- match.phylo.data(water02_tree, phy_water02)
match_asv_phylo_water02
```

### Phylogenetic alpha diversity

```{r phyalpha}
# Faith's phylogenetic diversity and standardized effect size
set.seed(1)

faith_pd_ses <- ses.pd(samp = (t(match_asv_phylo$data)), 
                       tree= pruned_tree,
                       null.model = "taxa.labels",
                       runs = 999)
head(faith_pd_ses)

# Standardized Effect Size of Mean Nearest Taxon Index (MNTD)
phy_dist <- cophenetic(match_asv_phylo$phy) #creating phylogenetic distance matrix
phy_dist

mntd_ses <- ses.mntd((t(match_asv_phylo$data)), phy_dist,
                          null.model = "taxa.labels",
                          abundance.weighted = F,
                          runs = 999) # calculate MNTD

nti <- as.matrix(-1 * mntd_ses[,6]) # calculate Nearest Taxon Index
rownames(nti) <- rownames(mntd_ses)
colnames(nti) <- "NTI"
head(nti)

mntd_nti_table <- cbind(mntd_ses, nti)

hist(nti) # check distribution
abline(v = 0, col = "red", lwd = 2, lty = 3)

# Tabulating alpha diversity values
data_scores_phy_alpha <- as.data.frame(cbind(faith_pd_ses$pd.obs, faith_pd_ses$pd.obs.z, mntd_ses$NTI))
colnames(data_scores_phy_alpha) <- c("PD", "PD.ses", "NTI")

data_scores_phy_alpha$type <- SRS_dna_meta$type
data_scores_phy_alpha$site_x_cage <- SRS_dna_meta$site_x_cage
data_scores_phy_alpha

# Reordering facets
data_scores_phy_alpha$type <- factor(data_scores_phy_alpha$type, 
                                 levels = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"))
data_scores_phy_alpha$site_x_cage <- factor(data_scores_phy_alpha$site_x_cage, 
                                        levels = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"))


# Rank sums and post hoc tests
kruskal.test(faith_pd_ses$pd.obs~SRS_dna_meta$site_x_cage) # Faith's PD by site
faith.dunn.site <- dunnTest(faith_pd_ses$pd.obs, as.factor(SRS_dna_meta$site_x_cage), method="bh", list = T)

faith_dunn_site <- as.data.frame(faith.dunn.site$res)
faith_dunn_site <- tibble::column_to_rownames(faith_dunn_site, var = "Comparison")
faith_dunn_site

kruskal.test(faith_pd_ses$pd.obs~SRS_dna_meta$cage_x_type) # Faith's PD by type
faith.dunn.type <- dunnTest(faith_pd_ses$pd.obs, as.factor(SRS_dna_meta$cage_x_type), method="bh")

faith_dunn_type <- as.data.frame(faith.dunn.type$res)
faith_dunn_type <- tibble::column_to_rownames(faith_dunn_type, var = "Comparison")
faith_dunn_type

nti_df <- as.data.frame(nti)

kruskal.test(nti_df$NTI~SRS_dna_meta$site_x_cage)
nti.dunn.site <- dunnTest(nti_df$NTI, as.factor(SRS_dna_meta$site_x_cage), method="bh", list = T) 

nti_dunn_site <- as.data.frame(nti.dunn.site$res)
nti_dunn_site <- tibble::column_to_rownames(nti_dunn_site, var = "Comparison")
nti_dunn_site


kruskal.test(nti_df$NTI~SRS_dna_meta$cage_x_type)
nti.dunn.type <- dunnTest(nti_df$NTI, as.factor(SRS_dna_meta$cage_x_type), method="bh", list = T) 

nti_dunn_type <- as.data.frame(nti.dunn.type$res)
nti_dunn_type <- tibble::column_to_rownames(nti_dunn_type, var = "Comparison")
nti_dunn_type
```

### Phylogenetic beta diversity

```{r phybeta}
# Phylogenetic dissimilarities (all)
set.seed(1)

wUniF <- UniFrac(physeq_tree, weighted = TRUE)
uUniF <- UniFrac(physeq_tree, weighted = FALSE)


# Phylogenetic dissimilarities between types at each site
## weighted UniFrac
####bremen
wUniF.bremen <- UniFrac(physeq_bremen, weighted = TRUE)

####brake
wUniF.brake <- UniFrac(physeq_brake, weighted = TRUE)

####bremerhaven
wUniF.bremerhaven <- UniFrac(physeq_bremerhaven, weighted = TRUE)

####helgoland
wUniF.helgoland <- UniFrac(physeq_helgoland, weighted = TRUE)

# Phylogenetic dissimilarities within each type across sites
## weighted UniFrac
####hdpe
wUniF.hdpe <- UniFrac(physeq_hdpe, weighted = TRUE)

####tyre wear
wUniF.tw <- UniFrac(physeq_tw, weighted = TRUE)

####wood
wUniF.wood <- UniFrac(physeq_wood, weighted = TRUE)

####water 3
wUniF.water3 <- UniFrac(physeq_water3, weighted = TRUE)

####water 02
wUniF.water02 <- UniFrac(physeq_water02, weighted = TRUE)


# Phylogenetic dissimilarities between types at each site
## unweighted UniFrac
####bremen
uUniF.bremen <- UniFrac(physeq_bremen, weighted = FALSE)

####brake
uUniF.brake <- UniFrac(physeq_brake, weighted = FALSE)

####bremerhaven
uUniF.bremerhaven <- UniFrac(physeq_bremerhaven, weighted = FALSE)

####helgoland
uUniF.helgoland <- UniFrac(physeq_helgoland, weighted = FALSE)

# Phylogenetic dissimilarities within each type across sites
## unweighted UniFrac
####hdpe
uUniF.hdpe <- UniFrac(physeq_hdpe, weighted = FALSE)

####tyre wear
uUniF.tw <- UniFrac(physeq_tw, weighted = FALSE)

####wood
uUniF.wood <- UniFrac(physeq_wood, weighted = FALSE)

####water 3
uUniF.water3 <- UniFrac(physeq_water3, weighted = FALSE)

####water 02
uUniF.water02 <- UniFrac(physeq_water02, weighted = FALSE)
```

### Principal Coordinate Analysis (PCoA)

```{r pcoa}
## Based on Bray Curtis (all)
PCoA_BC <- pcoa(dna_BC_matrix)
PCoA_BC$values
PCoA_BC$vectors

PCoA_BC_values <- as.data.frame(PCoA_BC$values)
PCoA_BC_vectors <- as.data.frame(PCoA_BC$vectors)
PCoA_BC_vectors$Axis.1 <- - PCoA_BC_vectors$Axis.1
PCoA_BC_vectors$type <- SRS_dna_meta$type
PCoA_BC_vectors$site_x_cage <- SRS_dna_meta$site_x_cage

PCoA_BC_vectors

barplot(PCoA_BC$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues') # distribution of eigenvalues

## Based on Jaccard (all)
PCoA_JC <- pcoa(dna_JC_matrix)
PCoA_JC$values
PCoA_JC$vectors

PCoA_JC_values <- as.data.frame(PCoA_JC$values)
PCoA_JC_vectors <- as.data.frame(PCoA_JC$vectors)
PCoA_JC_vectors$Axis.2 <- - PCoA_JC_vectors$Axis.2
PCoA_JC_vectors$type <- SRS_dna_meta$type
PCoA_JC_vectors$site_x_cage <- SRS_dna_meta$site_x_cage

PCoA_JC_vectors

barplot(PCoA_JC$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

## Based on weighted UniFrac (all)
PCoA_wUniF <- ordinate(physeq_tree, method = "PCoA", distance = wUniF)
PCoA_wUniF_vectors <- as.matrix(PCoA_wUniF$vectors)
PCoA_wUniF_values <- as.matrix(PCoA_wUniF$values)

PCoA_wUniF_values <- as.data.frame(PCoA_wUniF$values)
PCoA_wUniF_vectors <- as.data.frame(PCoA_wUniF$vectors)
PCoA_wUniF_vectors$site_x_cage <- SRS_dna_meta$site_x_cage
PCoA_wUniF_vectors$type <- SRS_dna_meta$type

barplot(PCoA_wUniF$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

## Based on unweighted UniFrac (all)
PCoA_uUniF <- ordinate(physeq_tree, method = "PCoA", distance = uUniF)
PCoA_uUniF_vectors <- as.matrix(PCoA_uUniF$vectors)
PCoA_uUniF_values <- as.matrix(PCoA_uUniF$values)

PCoA_uUniF_values <- as.data.frame(PCoA_uUniF$values)
PCoA_uUniF_vectors <- as.data.frame(PCoA_uUniF$vectors)
PCoA_uUniF_vectors$site_x_cage <- SRS_dna_meta$site_x_cage
PCoA_uUniF_vectors$type <- SRS_dna_meta$type

barplot(PCoA_uUniF$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

## Based on Bray Curtis by site
####bremen
PCoA_BC_bremen <- pcoa(bremen_BC_matrix)
PCoA_BC_bremen$values
PCoA_BC_bremen$vectors

PCoA_BC_bremen_values <- as.data.frame(PCoA_BC_bremen$values)
PCoA_BC_bremen_vectors <- as.data.frame(PCoA_BC_bremen$vectors)
PCoA_BC_bremen_vectors$Axis.2 <- - PCoA_BC_bremen_vectors$Axis.2 
PCoA_BC_bremen_vectors$type <- SRS_bremen$type
PCoA_BC_bremen_vectors$site_x_cage <- SRS_bremen$site_x_cage


barplot(PCoA_BC_bremen$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####brake
PCoA_BC_brake <- pcoa(brake_BC_matrix)
PCoA_BC_brake$values
PCoA_BC_brake$vectors
      
PCoA_BC_brake_values <- as.data.frame(PCoA_BC_brake$values)
PCoA_BC_brake_vectors <- as.data.frame(PCoA_BC_brake$vectors)
PCoA_BC_brake_vectors$type <- SRS_brake$type
PCoA_BC_brake_vectors$site_x_cage <- SRS_brake$site_x_cage

barplot(PCoA_BC_brake$values$Relative_eig, names = paste('PCoA'), las = 1, 
              ylab = 'Relative Eigenvalues')

####bremerhaven
PCoA_BC_bremerhaven <- pcoa(bremerhaven_BC_matrix)
PCoA_BC_bremerhaven$values
PCoA_BC_bremerhaven$vectors

PCoA_BC_bremerhaven_values <- as.data.frame(PCoA_BC_bremerhaven$values)
PCoA_BC_bremerhaven_vectors <- as.data.frame(PCoA_BC_bremerhaven$vectors)
PCoA_BC_bremerhaven_vectors$type <- SRS_bremerhaven$type
PCoA_BC_bremerhaven_vectors$site_x_cage <- SRS_bremerhaven$site_x_cage

barplot(PCoA_BC_bremerhaven$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####helgoland
PCoA_BC_helgoland <- pcoa(helgoland_BC_matrix)
PCoA_BC_helgoland$values
PCoA_BC_helgoland$vectors

PCoA_BC_helgoland_values <- as.data.frame(PCoA_BC_helgoland$values)
PCoA_BC_helgoland_vectors <- as.data.frame(PCoA_BC_helgoland$vectors)
PCoA_BC_helgoland_vectors$type <- SRS_helgoland$type
PCoA_BC_helgoland_vectors$cage <- SRS_helgoland$cage
PCoA_BC_helgoland_vectors$site_x_cage <- SRS_helgoland$site_x_cage

barplot(PCoA_BC_helgoland$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')


## Based on Bray Curtis by type
####hdpe
PCoA_BC_hdpe <- pcoa(hdpe_BC_matrix)
PCoA_BC_hdpe$values
PCoA_BC_hdpe$vectors
PCoA_BC_hdpe_values <- as.data.frame(PCoA_BC_hdpe$values)
PCoA_BC_hdpe_vectors <- as.data.frame(PCoA_BC_hdpe$vectors)
PCoA_BC_hdpe_vectors$site_x_cage <- SRS_hdpe$type
PCoA_BC_hdpe_vectors$site_x_cage <- SRS_hdpe$site_x_cage

barplot(PCoA_BC_hdpe$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####tyre wear
PCoA_BC_tw <- pcoa(tw_BC_matrix)
PCoA_BC_tw$values
PCoA_BC_tw$vectors
PCoA_BC_tw_values <- as.data.frame(PCoA_BC_tw$values)
PCoA_BC_tw_vectors <- as.data.frame(PCoA_BC_tw$vectors)
PCoA_BC_tw_vectors$site_x_cage <- SRS_tw$type
PCoA_BC_tw_vectors$site_x_cage <- SRS_tw$site_x_cage

barplot(PCoA_BC_tw$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####wood
PCoA_BC_wood <- pcoa(wood_BC_matrix)
PCoA_BC_wood$values
PCoA_BC_wood$vectors
PCoA_BC_wood_values <- as.data.frame(PCoA_BC_wood$values)
PCoA_BC_wood_vectors <- as.data.frame(PCoA_BC_wood$vectors)
PCoA_BC_wood_vectors$site_x_cage <- SRS_wood$type
PCoA_BC_wood_vectors$site_x_cage <- SRS_wood$site_x_cage

barplot(PCoA_BC_wood$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####water 3
PCoA_BC_water3 <- pcoa(water3_BC_matrix)
PCoA_BC_water3$values
PCoA_BC_water3$vectors
PCoA_BC_water3_values <- as.data.frame(PCoA_BC_water3$values)
PCoA_BC_water3_vectors <- as.data.frame(PCoA_BC_water3$vectors)
PCoA_BC_water3_vectors$site_x_cage <- SRS_water3$type
PCoA_BC_water3_vectors$site_x_cage <- SRS_water3$site_x_cage

barplot(PCoA_BC_water3$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####water 02
PCoA_BC_water02 <- pcoa(water02_BC_matrix)
PCoA_BC_water02$values
PCoA_BC_water02$vectors
PCoA_BC_water02_values <- as.data.frame(PCoA_BC_water02$values)
PCoA_BC_water02_vectors <- as.data.frame(PCoA_BC_water02$vectors)
PCoA_BC_water02_vectors$site_x_cage <- SRS_water02$type
PCoA_BC_water02_vectors$site_x_cage <- SRS_water02$site_x_cage

barplot(PCoA_BC_water02$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')


## Based on Jaccard by site
####bremen
PCoA_JC_bremen <- pcoa(bremen_JC_matrix)
PCoA_JC_bremen$values
PCoA_JC_bremen$vectors
PCoA_JC_bremen_values <- as.data.frame(PCoA_JC_bremen$values)
PCoA_JC_bremen_vectors <- as.data.frame(PCoA_JC_bremen$vectors)
PCoA_JC_bremen_vectors$Axis.2 <- - PCoA_JC_bremen_vectors$Axis.2
PCoA_JC_bremen_vectors$type <- SRS_bremen$type
PCoA_JC_bremen_vectors$site_x_cage <- SRS_bremen$site_x_cage

barplot(PCoA_JC_bremen$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####brake
PCoA_JC_brake <- pcoa(brake_JC_matrix)
PCoA_JC_brake$values
PCoA_JC_brake$vectors
PCoA_JC_brake_values <- as.data.frame(PCoA_JC_brake$values)
PCoA_JC_brake_vectors <- as.data.frame(PCoA_JC_brake$vectors)
PCoA_JC_brake_vectors$Axis.1 <- - PCoA_JC_brake_vectors$Axis.1
PCoA_JC_brake_vectors$type <- SRS_brake$type
PCoA_JC_brake_vectors$site_x_cage <- SRS_brake$site_x_cage

barplot(PCoA_JC_brake$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####bremerhaven
PCoA_JC_bremerhaven <- pcoa(bremerhaven_JC_matrix)
PCoA_JC_bremerhaven$values
PCoA_JC_bremerhaven$vectors
PCoA_JC_bremerhaven_values <- as.data.frame(PCoA_JC_bremerhaven$values)
PCoA_JC_bremerhaven_vectors <- as.data.frame(PCoA_JC_bremerhaven$vectors)
PCoA_JC_bremerhaven_vectors$Axis.2 <- - PCoA_JC_bremerhaven_vectors$Axis.2
PCoA_JC_bremerhaven_vectors$type <- SRS_bremerhaven$type
PCoA_JC_bremerhaven_vectors$site_x_cage <- SRS_bremerhaven$site_x_cage

barplot(PCoA_JC_bremerhaven$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####helgoland
PCoA_JC_helgoland <- pcoa(helgoland_JC_matrix)
PCoA_JC_helgoland$values
PCoA_JC_helgoland$vectors
PCoA_JC_helgoland_values <- as.data.frame(PCoA_JC_helgoland$values)
PCoA_JC_helgoland_vectors <- as.data.frame(PCoA_JC_helgoland$vectors)
PCoA_JC_helgoland_vectors$Axis.2 <- - PCoA_JC_helgoland_vectors$Axis.2
PCoA_JC_helgoland_vectors$type <- SRS_helgoland$type
PCoA_JC_helgoland_vectors$cage <- SRS_helgoland$cage
PCoA_JC_helgoland_vectors$site_x_cage <- SRS_helgoland$site_x_cage

barplot(PCoA_JC_helgoland$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

## Based on Jaccard by type
####hdpe
PCoA_JC_hdpe <- pcoa(hdpe_JC_matrix)
PCoA_JC_hdpe$values
PCoA_JC_hdpe$vectors
PCoA_JC_hdpe_values <- as.data.frame(PCoA_JC_hdpe$values)
PCoA_JC_hdpe_vectors <- as.data.frame(PCoA_JC_hdpe$vectors)
PCoA_JC_hdpe_vectors$type <- SRS_hdpe$type
PCoA_JC_hdpe_vectors$site_x_cage <- SRS_hdpe$site_x_cage

barplot(PCoA_JC_hdpe$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####tyre wear
PCoA_JC_tw <- pcoa(tw_JC_matrix)
PCoA_JC_tw$values
PCoA_JC_tw$vectors
PCoA_JC_tw_values <- as.data.frame(PCoA_JC_tw$values)
PCoA_JC_tw_vectors <- as.data.frame(PCoA_JC_tw$vectors)
PCoA_JC_tw_vectors$type <- SRS_tw$type
PCoA_JC_tw_vectors$site_x_cage <- SRS_tw$site_x_cage

barplot(PCoA_JC_tw$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####wood
PCoA_JC_wood <- pcoa(wood_JC_matrix)
PCoA_JC_wood$values
PCoA_JC_wood$vectors
PCoA_JC_wood_values <- as.data.frame(PCoA_JC_wood$values)
PCoA_JC_wood_vectors <- as.data.frame(PCoA_JC_wood$vectors)
PCoA_JC_wood_vectors$type <- SRS_wood$type
PCoA_JC_wood_vectors$site_x_cage <- SRS_wood$site_x_cage

barplot(PCoA_JC_wood$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####water 3
PCoA_JC_water3 <- pcoa(water3_JC_matrix)
PCoA_JC_water3$values
PCoA_JC_water3$vectors
PCoA_JC_water3_values <- as.data.frame(PCoA_JC_water3$values)
PCoA_JC_water3_vectors <- as.data.frame(PCoA_JC_water3$vectors)
PCoA_JC_water3_vectors$Axis.2 <- - PCoA_JC_water3_vectors$Axis.2
PCoA_JC_water3_vectors$type <- SRS_water3$type
PCoA_JC_water3_vectors$site_x_cage <- SRS_water3$site_x_cage

barplot(PCoA_JC_water3$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####water 02
PCoA_JC_water02 <- pcoa(water02_JC_matrix)
PCoA_JC_water02$values
PCoA_JC_water02$vectors

PCoA_JC_water02_values <- as.data.frame(PCoA_JC_water02$values)
PCoA_JC_water02_vectors <- as.data.frame(PCoA_JC_water02$vectors)
PCoA_JC_water02_vectors$type <- SRS_water02$type
PCoA_JC_water02_vectors$site_x_cage <- SRS_water02$site_x_cage

barplot(PCoA_JC_water02$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')


## Based on weighted UniFrac by site
####bremen
PCoA_wUniF_bremen <- ordinate(physeq_bremen, method = "PCoA", distance = wUniF.bremen)
PCoA_wUniF_vectors_bremen <- as.matrix(PCoA_wUniF_bremen$vectors)
PCoA_wUniF_values_bremen <- as.matrix(PCoA_wUniF_bremen$values)


PCoA_wUniF_values_bremen <- as.data.frame(PCoA_wUniF_bremen$values)
PCoA_wUniF_vectors_bremen <- as.data.frame(PCoA_wUniF_bremen$vectors)
PCoA_wUniF_vectors_bremen$Axis.1 <- - PCoA_wUniF_vectors_bremen$Axis.1
PCoA_wUniF_vectors_bremen$site_x_cage <- SRS_bremen$site_x_cage
PCoA_wUniF_vectors_bremen$type <- SRS_bremen$type

barplot(PCoA_wUniF_bremen$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####brake
PCoA_wUniF_brake <- ordinate(physeq_brake, method = "PCoA", distance = wUniF.brake)
PCoA_wUniF_vectors_brake <- as.matrix(PCoA_wUniF_brake$vectors)
PCoA_wUniF_values_brake <- as.matrix(PCoA_wUniF_brake$values)

PCoA_wUniF_values_brake <- as.data.frame(PCoA_wUniF_brake$values)
PCoA_wUniF_vectors_brake <- as.data.frame(PCoA_wUniF_brake$vectors)
PCoA_wUniF_vectors_brake$Axis.1 <- - PCoA_wUniF_vectors_brake$Axis.1
PCoA_wUniF_vectors_brake$site_x_cage <- SRS_brake$site_x_cage
PCoA_wUniF_vectors_brake$type <- SRS_brake$type

barplot(PCoA_wUniF_brake$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####bremerhaven
PCoA_wUniF_bremerhaven <- ordinate(physeq_bremerhaven, method = "PCoA", distance = wUniF.bremerhaven)
PCoA_wUniF_vectors_bremerhaven <- as.matrix(PCoA_wUniF_bremerhaven$vectors)
PCoA_wUniF_values_bremerhaven <- as.matrix(PCoA_wUniF_bremerhaven$values)

PCoA_wUniF_values_bremerhaven <- as.data.frame(PCoA_wUniF_bremerhaven$values)
PCoA_wUniF_vectors_bremerhaven <- as.data.frame(PCoA_wUniF_bremerhaven$vectors)
PCoA_wUniF_vectors_bremerhaven$site_x_cage <- SRS_bremerhaven$site_x_cage
PCoA_wUniF_vectors_bremerhaven$type <- SRS_bremerhaven$type

barplot(PCoA_wUniF_bremerhaven$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####helgoland
PCoA_wUniF_helgoland <- ordinate(physeq_helgoland, method = "PCoA", distance = wUniF.helgoland)
PCoA_wUniF_vectors_helgoland <- as.matrix(PCoA_wUniF_helgoland$vectors)
PCoA_wUniF_values_helgoland <- as.matrix(PCoA_wUniF_helgoland$values)


PCoA_wUniF_values_helgoland <- as.data.frame(PCoA_wUniF_helgoland$values)

PCoA_wUniF_vectors_helgoland <- as.data.frame(PCoA_wUniF_helgoland$vectors)
PCoA_wUniF_vectors_helgoland$Axis.1 <- - PCoA_wUniF_vectors_helgoland$Axis.1
PCoA_wUniF_vectors_helgoland$Axis.2 <- - PCoA_wUniF_vectors_helgoland$Axis.2
PCoA_wUniF_vectors_helgoland$cage <- SRS_helgoland$cage
PCoA_wUniF_vectors_helgoland$site_x_cage <- SRS_helgoland$site_x_cage
PCoA_wUniF_vectors_helgoland$type <- SRS_helgoland$type

barplot(PCoA_wUniF_helgoland$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')


## Based on weighted UniFrac by type
####hdpe
PCoA_wUniF_hdpe <- ordinate(physeq_hdpe, method = "PCoA", distance = wUniF.hdpe)
PCoA_wUniF_vectors_hdpe <- as.matrix(PCoA_wUniF_hdpe$vectors)
PCoA_wUniF_values_hdpe <- as.matrix(PCoA_wUniF_hdpe$values)


PCoA_wUniF_values_hdpe <- as.data.frame(PCoA_wUniF_hdpe$values)
PCoA_wUniF_vectors_hdpe <- as.data.frame(PCoA_wUniF_hdpe$vectors)
PCoA_wUniF_vectors_hdpe$site_x_cage <- SRS_hdpe$site_x_cage
PCoA_wUniF_vectors_hdpe$type <- SRS_hdpe$type

barplot(PCoA_wUniF_hdpe$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####tw
PCoA_wUniF_tw <- ordinate(physeq_tw, method = "PCoA", distance = wUniF.tw)
PCoA_wUniF_vectors_tw <- as.matrix(PCoA_wUniF_tw$vectors)
PCoA_wUniF_values_tw <- as.matrix(PCoA_wUniF_tw$values)


PCoA_wUniF_values_tw <- as.data.frame(PCoA_wUniF_tw$values)
PCoA_wUniF_vectors_tw <- as.data.frame(PCoA_wUniF_tw$vectors)
PCoA_wUniF_vectors_tw$site_x_cage <- SRS_tw$site_x_cage
PCoA_wUniF_vectors_tw$type <- SRS_tw$type

barplot(PCoA_wUniF_tw$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####wood
PCoA_wUniF_wood <- ordinate(physeq_wood, method = "PCoA", distance = wUniF.wood)
PCoA_wUniF_vectors_wood <- as.matrix(PCoA_wUniF_wood$vectors)
PCoA_wUniF_values_wood <- as.matrix(PCoA_wUniF_wood$values)


PCoA_wUniF_values_wood <- as.data.frame(PCoA_wUniF_wood$values)
PCoA_wUniF_vectors_wood <- as.data.frame(PCoA_wUniF_wood$vectors)
PCoA_wUniF_vectors_wood$site_x_cage <- SRS_wood$site_x_cage
PCoA_wUniF_vectors_wood$type <- SRS_wood$type

barplot(PCoA_wUniF_wood$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####water 3
PCoA_wUniF_water3 <- ordinate(physeq_water3, method = "PCoA", distance = wUniF.water3)
PCoA_wUniF_vectors_water3 <- as.matrix(PCoA_wUniF_water3$vectors)
PCoA_wUniF_values_water3 <- as.matrix(PCoA_wUniF_water3$values)


PCoA_wUniF_values_water3 <- as.data.frame(PCoA_wUniF_water3$values)
PCoA_wUniF_vectors_water3 <- as.data.frame(PCoA_wUniF_water3$vectors)
PCoA_wUniF_vectors_water3$Axis.1 <- - PCoA_wUniF_vectors_water3$Axis.1
PCoA_wUniF_vectors_water3$site_x_cage <- SRS_water3$site_x_cage
PCoA_wUniF_vectors_water3$type <- SRS_water3$type

barplot(PCoA_wUniF_water3$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

#### water02
PCoA_wUniF_water02 <- ordinate(physeq_water02, method = "PCoA", distance = wUniF.water02)
PCoA_wUniF_vectors_water02 <- as.matrix(PCoA_wUniF_water02$vectors)
PCoA_wUniF_values_water02 <- as.matrix(PCoA_wUniF_water02$values)

PCoA_wUniF_values_water02 <- as.data.frame(PCoA_wUniF_water02$values)
PCoA_wUniF_vectors_water02 <- as.data.frame(PCoA_wUniF_water02$vectors)
PCoA_wUniF_vectors_water02$site_x_cage <- SRS_water02$site_x_cage
PCoA_wUniF_vectors_water02$type <- SRS_water02$type

barplot(PCoA_wUniF_water02$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')


## Based on unweighted UniFrac by site
####bremen
PCoA_uUniF_bremen <- ordinate(physeq_bremen, method = "PCoA", distance = uUniF.bremen)
PCoA_uUniF_vectors_bremen <- as.matrix(PCoA_uUniF_bremen$vectors)
PCoA_uUniF_values_bremen <- as.matrix(PCoA_uUniF_bremen$values)


PCoA_uUniF_values_bremen <- as.data.frame(PCoA_uUniF_bremen$values)
PCoA_uUniF_vectors_bremen <- as.data.frame(PCoA_uUniF_bremen$vectors)
PCoA_uUniF_vectors_bremen$Axis.2 <- - PCoA_uUniF_vectors_bremen$Axis.2
PCoA_uUniF_vectors_bremen$site_x_cage <- SRS_bremen$site_x_cage
PCoA_uUniF_vectors_bremen$type <- SRS_bremen$type

barplot(PCoA_uUniF_bremen$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####brake
PCoA_uUniF_brake <- ordinate(physeq_brake, method = "PCoA", distance = uUniF.brake)
PCoA_uUniF_vectors_brake <- as.matrix(PCoA_uUniF_brake$vectors)
PCoA_uUniF_values_brake <- as.matrix(PCoA_uUniF_brake$values)

PCoA_uUniF_values_brake <- as.data.frame(PCoA_uUniF_brake$values)
PCoA_uUniF_vectors_brake <- as.data.frame(PCoA_uUniF_brake$vectors)
PCoA_uUniF_vectors_brake$Axis.1 <- - PCoA_uUniF_vectors_brake$Axis.1
PCoA_uUniF_vectors_brake$site_x_cage <- SRS_brake$site_x_cage
PCoA_uUniF_vectors_brake$type <- SRS_brake$type

barplot(PCoA_uUniF_brake$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####bremerhaven
PCoA_uUniF_bremerhaven <- ordinate(physeq_bremerhaven, method = "PCoA", distance = uUniF.bremerhaven)
PCoA_uUniF_vectors_bremerhaven <- as.matrix(PCoA_uUniF_bremerhaven$vectors)
PCoA_uUniF_values_bremerhaven <- as.matrix(PCoA_uUniF_bremerhaven$values)


PCoA_uUniF_values_bremerhaven <- as.data.frame(PCoA_uUniF_bremerhaven$values)
PCoA_uUniF_vectors_bremerhaven <- as.data.frame(PCoA_uUniF_bremerhaven$vectors)
PCoA_uUniF_vectors_bremerhaven$Axis.2 <- - PCoA_uUniF_vectors_bremerhaven$Axis.2
PCoA_uUniF_vectors_bremerhaven$site_x_cage <- SRS_bremerhaven$site_x_cage
PCoA_uUniF_vectors_bremerhaven$type <- SRS_bremerhaven$type

barplot(PCoA_uUniF_bremerhaven$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####helgoland
PCoA_uUniF_helgoland <- ordinate(physeq_helgoland, method = "PCoA", distance = uUniF.helgoland)
PCoA_uUniF_vectors_helgoland <- as.matrix(PCoA_uUniF_helgoland$vectors)
PCoA_uUniF_values_helgoland <- as.matrix(PCoA_uUniF_helgoland$values)


PCoA_uUniF_values_helgoland <- as.data.frame(PCoA_uUniF_helgoland$values)

PCoA_uUniF_vectors_helgoland <- as.data.frame(PCoA_uUniF_helgoland$vectors)
PCoA_uUniF_vectors_helgoland$Axis.2 <- - PCoA_uUniF_vectors_helgoland$Axis.2
PCoA_uUniF_vectors_helgoland$cage <- SRS_helgoland$cage
PCoA_uUniF_vectors_helgoland$site_x_cage <- SRS_helgoland$site_x_cage
PCoA_uUniF_vectors_helgoland$type <- SRS_helgoland$type

barplot(PCoA_uUniF_helgoland$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

## Based on unweighted UniFrac by type
####hdpe
PCoA_uUniF_hdpe <- ordinate(physeq_hdpe, method = "PCoA", distance = uUniF.hdpe)
PCoA_uUniF_vectors_hdpe <- as.matrix(PCoA_uUniF_hdpe$vectors)
PCoA_uUniF_values_hdpe <- as.matrix(PCoA_uUniF_hdpe$values)


PCoA_uUniF_values_hdpe <- as.data.frame(PCoA_uUniF_hdpe$values)
PCoA_uUniF_vectors_hdpe <- as.data.frame(PCoA_uUniF_hdpe$vectors)
PCoA_uUniF_vectors_hdpe$Axis.2 <- - PCoA_uUniF_vectors_hdpe$Axis.2
PCoA_uUniF_vectors_hdpe$site_x_cage <- SRS_hdpe$site_x_cage
PCoA_uUniF_vectors_hdpe$type <- SRS_hdpe$type

barplot(PCoA_uUniF_hdpe$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####tw
PCoA_uUniF_tw <- ordinate(physeq_tw, method = "PCoA", distance = uUniF.tw)
PCoA_uUniF_vectors_tw <- as.matrix(PCoA_uUniF_tw$vectors)
PCoA_uUniF_values_tw <- as.matrix(PCoA_uUniF_tw$values)


PCoA_uUniF_values_tw <- as.data.frame(PCoA_uUniF_tw$values)
PCoA_uUniF_vectors_tw <- as.data.frame(PCoA_uUniF_tw$vectors)
PCoA_uUniF_vectors_tw$site_x_cage <- SRS_tw$site_x_cage
PCoA_uUniF_vectors_tw$type <- SRS_tw$type

barplot(PCoA_uUniF_tw$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####wood
PCoA_uUniF_wood <- ordinate(physeq_wood, method = "PCoA", distance = uUniF.wood)
PCoA_uUniF_vectors_wood <- as.matrix(PCoA_uUniF_wood$vectors)
PCoA_uUniF_values_wood <- as.matrix(PCoA_uUniF_wood$values)


PCoA_uUniF_values_wood <- as.data.frame(PCoA_uUniF_wood$values)
PCoA_uUniF_vectors_wood <- as.data.frame(PCoA_uUniF_wood$vectors)
PCoA_uUniF_vectors_wood$site_x_cage <- SRS_wood$site_x_cage
PCoA_uUniF_vectors_wood$type <- SRS_wood$type

barplot(PCoA_uUniF_wood$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

####water 3
PCoA_uUniF_water3 <- ordinate(physeq_water3, method = "PCoA", distance = uUniF.water3)
PCoA_uUniF_vectors_water3 <- as.matrix(PCoA_uUniF_water3$vectors)
PCoA_uUniF_values_water3 <- as.matrix(PCoA_uUniF_water3$values)


PCoA_uUniF_values_water3 <- as.data.frame(PCoA_uUniF_water3$values)
PCoA_uUniF_vectors_water3 <- as.data.frame(PCoA_uUniF_water3$vectors)
PCoA_uUniF_vectors_water3$site_x_cage <- SRS_water3$site_x_cage
PCoA_uUniF_vectors_water3$type <- SRS_water3$type

barplot(PCoA_uUniF_water3$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

#### water02
PCoA_uUniF_water02 <- ordinate(physeq_water02, method = "PCoA", distance = uUniF.water02)
PCoA_uUniF_vectors_water02 <- as.matrix(PCoA_uUniF_water02$vectors)
PCoA_uUniF_values_water02 <- as.matrix(PCoA_uUniF_water02$values)

PCoA_uUniF_values_water02 <- as.data.frame(PCoA_uUniF_water02$values)
PCoA_uUniF_vectors_water02 <- as.data.frame(PCoA_uUniF_water02$vectors)
PCoA_uUniF_vectors_water02$site_x_cage <- SRS_water02$site_x_cage
PCoA_uUniF_vectors_water02$type <- SRS_water02$type

barplot(PCoA_uUniF_water02$values$Relative_eig, names = paste('PCoA'), las = 1, 
        ylab = 'Relative Eigenvalues')

```

### Beta partitioning
Partitioning taxonomic and phylogenetic dissimilarities into components of turnover and nestedness based on the Jaccard index
```{r partitioning}
# the mean count of triplicate samples for each type were calculated and imported 
####hdpe
mean_hdpe <- read.csv("mean_SRS_hdpe.csv", sep = "\t", dec = ".", header = T, row.names = 1)

bin_hdpe <- as.data.frame(mean_hdpe) # make binary
bin_hdpe[bin_hdpe > 0] <- 1
bin_hdpe <- t(bin_hdpe)
dim(bin_hdpe)

# partition dissimilarities
core.hdpe <- betapart.core(bin_hdpe) # summary
core.hdpe

multi.hdpe <- beta.multi(core.hdpe, index.family = "jaccard") # global
multi.hdpe

pair.hdpe <- beta.pair(core.hdpe, index.family = "jaccard") # pairwise comparison by site
pair.hdpe

hdpe.beta.jtu <- as.matrix(pair.hdpe$beta.jtu)
hdpe.beta.jne <- as.matrix(pair.hdpe$beta.jne)
hdpe.beta.jac <- as.matrix(pair.hdpe$beta.jac)

write.table(hdpe.beta.jtu, "hdpe_beta_jtu.txt", sep = "\t")
write.table(hdpe.beta.jne, "hdpe_beta_jne.txt", sep = "\t")
write.table(hdpe.beta.jac, "hdpe_beta_jac.txt", sep = "\t")


####tw
mean_tw <- read.csv("mean_SRS_tw.csv", sep = "\t", dec = ".", header = T, row.names = 1)

bin_tw <- as.data.frame(mean_tw) # make binary
bin_tw[bin_tw > 0] <- 1
bin_tw <- t(bin_tw)
dim(bin_tw)

# partition dissimilarities
core.tw <- betapart.core(bin_tw) # summary
core.tw

multi.tw <- beta.multi(core.tw, index.family = "jaccard") # global
multi.tw

pair.tw <- beta.pair(core.tw, index.family = "jaccard") # pairwise comparison by site
pair.tw

tw.beta.jtu <- as.matrix(pair.tw$beta.jtu)
tw.beta.jne <- as.matrix(pair.tw$beta.jne)
tw.beta.jac <- as.matrix(pair.tw$beta.jac)

write.table(tw.beta.jtu, "tw_beta_jtu.txt", sep = "\t")
write.table(tw.beta.jne, "tw_beta_jne.txt", sep = "\t")
write.table(tw.beta.jac, "tw_beta_jac.txt", sep = "\t")


####wood
mean_wood <- read.csv("mean_SRS_wood.csv", sep = "\t", dec = ".", header = T, row.names = 1)

bin_wood <- as.data.frame(mean_wood) # make binary
bin_wood[bin_wood > 0] <- 1
bin_wood <- t(bin_wood)
dim(bin_wood)

# partition dissimilarities
core.wood <- betapart.core(bin_wood) # summary
core.wood

multi.wood <- beta.multi(core.wood, index.family = "jaccard") # global
multi.wood

pair.wood <- beta.pair(core.wood, index.family = "jaccard") # pairwise comparison by site
pair.wood

wood.beta.jtu <- as.matrix(pair.wood$beta.jtu)
wood.beta.jne <- as.matrix(pair.wood$beta.jne)
wood.beta.jac <- as.matrix(pair.wood$beta.jac)

write.table(wood.beta.jtu, "wood_beta_jtu.txt", sep = "\t")
write.table(wood.beta.jne, "wood_beta_jne.txt", sep = "\t")
write.table(wood.beta.jac, "wood_beta_jac.txt", sep = "\t")


####water3
mean_water3 <- read.csv("mean_SRS_water3.csv", sep = "\t", dec = ".", header = T, row.names = 1)

bin_water3 <- as.data.frame(mean_water3) # make binary
bin_water3[bin_water3 > 0] <- 1
bin_water3 <- t(bin_water3)
dim(bin_water3)

# partition dissimilarities
core.water3 <- betapart.core(bin_water3) # summary
core.water3

multi.water3 <- beta.multi(core.water3, index.family = "jaccard") # global
multi.water3

pair.water3 <- beta.pair(core.water3, index.family = "jaccard") # pairwise comparison by site
pair.water3

water3.beta.jtu <- as.matrix(pair.water3$beta.jtu)
water3.beta.jne <- as.matrix(pair.water3$beta.jne)
water3.beta.jac <- as.matrix(pair.water3$beta.jac)

write.table(water3.beta.jtu, "water3_beta_jtu.txt", sep = "\t")
write.table(water3.beta.jne, "water3_beta_jne.txt", sep = "\t")
write.table(water3.beta.jac, "water3_beta_jac.txt", sep = "\t")


####water02
mean_water02 <- read.csv("mean_SRS_water02.csv", sep = "\t", dec = ".", header = T, row.names = 1)

bin_water02 <- as.data.frame(mean_water02) # make binary
bin_water02[bin_water02 > 0] <- 1
bin_water02 <- t(bin_water02)
dim(bin_water02)

# partition dissimilarities
core.water02 <- betapart.core(bin_water02) # summary
core.water02

multi.water02 <- beta.multi(core.water02, index.family = "jaccard") # global
multi.water02

pair.water02 <- beta.pair(core.water02, index.family = "jaccard") # pairwise comparison by site
pair.water02

water02.beta.jtu <- as.matrix(pair.water02$beta.jtu)
water02.beta.jne <- as.matrix(pair.water02$beta.jne)
water02.beta.jac <- as.matrix(pair.water02$beta.jac)

write.table(water02.beta.jtu, "water02_beta_jtu.txt", sep = "\t")
write.table(water02.beta.jne, "water02_beta_jne.txt", sep = "\t")
write.table(water02.beta.jac, "water02_beta_jac.txt", sep = "\t")
```

### Quantitative Process Estimates (Community Assembly)
Beta NTI calculations were performed using script from Stegen et al. (2013): <https://github.com/stegen/Stegen_etal_ISME_2013/blob/master/bNTI_Local_Machine.r>
RCBray calculations were performed using the script from Franz-Sebastian Krah (2019): <https://github.com/FranzKrah/raup_crick/blob/master/raup_crick_abu_par.r>

```{r nullmodels}
# betaMNTD by type
####hdpe
bmntd_w_hdpe <- as.matrix(comdistnt(t(match_asv_phylo_hdpe$data), cophenetic(match_asv_phylo_hdpe$phy), 
                                   abundance.weighted = T))

dim(bmntd_w_hdpe)
bmntd_w_hdpe[1:5, 1:5]

#make sure names match, should both be true
identical(colnames(match_asv_phylo_hdpe$data), colnames(bmntd_w_hdpe))
identical(colnames(match_asv_phylo_hdpe$data), rownames(bmntd_w_hdpe))


# calculate randomized betaMNTD (tip randomization)
beta.reps = 999

rand_w_bmntd_comp <- array(c(-999),dim=c(ncol(match_asv_phylo_hdpe$data),
                                         ncol(match_asv_phylo_hdpe$data),
                                         beta.reps))
dim(rand_w_bmntd_comp)

for (rep in 1:beta.reps) {
        
        rand_w_bmntd_comp[,,rep] = as.matrix(comdistnt(t(match_asv_phylo_hdpe$data),
                                                       taxaShuffle(cophenetic(match_asv_phylo_hdpe$phy)),
                                                       abundance.weighted=T,
                                                       exclude.conspecifics = F));
        
        print(c(date(),rep));
        
}


# weighted betaNTI
weighted_bnti_hdpe <- matrix(c(NA), nrow = ncol(match_asv_phylo_hdpe$data),
                        ncol = ncol(match_asv_phylo_hdpe$data))
dim(weighted_bnti_hdpe)


for (columns in 1:(ncol(match_asv_phylo_hdpe$data)-1)) {
        for (rows in (columns+1):ncol(match_asv_phylo_hdpe$data)) {
                
                rand.vals = rand_w_bmntd_comp[rows,columns,];
                weighted_bnti_hdpe[rows,columns] = (bmntd_w_hdpe[rows,columns] - mean(rand.vals)) / sd(rand.vals);
                rm("rand.vals");
                
        };
};

rownames(weighted_bnti_hdpe) = colnames(match_asv_phylo_hdpe$data)
colnames(weighted_bnti_hdpe) = colnames(match_asv_phylo_hdpe$data)

weighted_bnti_hdpe
hist(weighted_bnti_hdpe)
abline(v = c(-2,2), col = "red", lwd = c(2,2), lty = c(3,3))


####tyre wear
bmntd_w_tw <- as.matrix(comdistnt(t(match_asv_phylo_tw$data), cophenetic(match_asv_phylo_tw$phy), 
                                    abundance.weighted = T))
dim(bmntd_w_tw)
bmntd_w_tw[1:5, 1:5]

# make sure names match, should both be true
identical(colnames(match_asv_phylo_tw$data), colnames(bmntd_w_tw))
identical(colnames(match_asv_phylo_tw$data), rownames(bmntd_w_tw))

# calculate randomized betaMNTD (tip randomization)
beta.reps = 999

rand_w_bmntd_comp <- array(c(-999),dim=c(ncol(match_asv_phylo_tw$data),
                                         ncol(match_asv_phylo_tw$data),
                                         beta.reps))
dim(rand_w_bmntd_comp)

for (rep in 1:beta.reps) {
        
        rand_w_bmntd_comp[,,rep] = as.matrix(comdistnt(t(match_asv_phylo_tw$data),
                                                       taxaShuffle(cophenetic(match_asv_phylo_tw$phy)),
                                                       abundance.weighted=T,
                                                       exclude.conspecifics = F));
        
        print(c(date(),rep));
        
}


# weighted betaNTI
weighted_bnti_tw <- matrix(c(NA), nrow = ncol(match_asv_phylo_tw$data),
                             ncol = ncol(match_asv_phylo_tw$data))
dim(weighted_bnti_tw)


for (columns in 1:(ncol(match_asv_phylo_tw$data)-1)) {
        for (rows in (columns+1):ncol(match_asv_phylo_tw$data)) {
                
                rand.vals = rand_w_bmntd_comp[rows,columns,];
                weighted_bnti_tw[rows,columns] = (bmntd_w_tw[rows,columns] - mean(rand.vals)) / sd(rand.vals);
                rm("rand.vals");
                
        };
};

rownames(weighted_bnti_tw) = colnames(match_asv_phylo_tw$data)
colnames(weighted_bnti_tw) = colnames(match_asv_phylo_tw$data)

weighted_bnti_tw
hist(weighted_bnti_tw)
abline(v = c(-2,2), col = "red", lwd = c(2,2), lty = c(3,3))

write.table(weighted_bnti_tw,"weighted_bnti_tw.txt", sep="\t")


####wood
bmntd_w_wood <- as.matrix(comdistnt(t(match_asv_phylo_wood$data), cophenetic(match_asv_phylo_wood$phy), 
                                    abundance.weighted = T))
dim(bmntd_w_wood)
bmntd_w_wood[1:5, 1:5]

# make sure names match, should both be true
identical(colnames(match_asv_phylo_wood$data), colnames(bmntd_w_wood))
identical(colnames(match_asv_phylo_wood$data), rownames(bmntd_w_wood))

# calculate randomized betaMNTD (tip randomization)
beta.reps = 999

rand_w_bmntd_comp <- array(c(-999),dim=c(ncol(match_asv_phylo_wood$data),
                                         ncol(match_asv_phylo_wood$data),
                                         beta.reps))
dim(rand_w_bmntd_comp)

for (rep in 1:beta.reps) {
        
        rand_w_bmntd_comp[,,rep] = as.matrix(comdistnt(t(match_asv_phylo_wood$data),
                                                       taxaShuffle(cophenetic(match_asv_phylo_wood$phy)),
                                                       abundance.weighted=T,
                                                       exclude.conspecifics = F));
        
        print(c(date(),rep));
        
}


# weighted betaNTI
weighted_bnti_wood <- matrix(c(NA), nrow = ncol(match_asv_phylo_wood$data),
                             ncol = ncol(match_asv_phylo_wood$data))
dim(weighted_bnti_wood)


for (columns in 1:(ncol(match_asv_phylo_wood$data)-1)) {
        for (rows in (columns+1):ncol(match_asv_phylo_wood$data)) {
                
                rand.vals = rand_w_bmntd_comp[rows,columns,];
                weighted_bnti_wood[rows,columns] = (bmntd_w_wood[rows,columns] - mean(rand.vals)) / sd(rand.vals);
                rm("rand.vals");
                
        };
};

rownames(weighted_bnti_wood) = colnames(match_asv_phylo_wood$data)
colnames(weighted_bnti_wood) = colnames(match_asv_phylo_wood$data)

weighted_bnti_wood
hist(weighted_bnti_wood)
abline(v = c(-2,2), col = "red", lwd = c(2,2), lty = c(3,3))


####water 3
bmntd_water3 <- as.matrix(comdistnt(t(match_asv_phylo_water3$data), cophenetic(match_asv_phylo_water3$phy), 
                                    abundance.weighted = T))
dim(bmntd_water3)
bmntd_water3[1:5, 1:5]

#make sure names match, should both be true
identical(colnames(match_asv_phylo_water3$data), colnames(bmntd_water3))
identical(colnames(match_asv_phylo_water3$data), rownames(bmntd_water3))

#calculate randomized betaMNTD (tip randomization)
beta.reps = 999

rand_w_bmntd_comp <- array(c(-999),dim=c(ncol(match_asv_phylo_water3$data),
                                         ncol(match_asv_phylo_water3$data),
                                         beta.reps))
dim(rand_w_bmntd_comp)

for (rep in 1:beta.reps) {
        
        rand_w_bmntd_comp[,,rep] = as.matrix(comdistnt(t(match_asv_phylo_water3$data),
                                                       taxaShuffle(cophenetic(match_asv_phylo_water3$phy)),
                                                       abundance.weighted=T,
                                                       exclude.conspecifics = F));
        
        print(c(date(),rep));
        
}


#weighted betaNTI
weighted_bnti_water3 <- matrix(c(NA), nrow = ncol(match_asv_phylo_water3$data),
                             ncol = ncol(match_asv_phylo_water3$data))
dim(weighted_bnti_water3)


for (columns in 1:(ncol(match_asv_phylo_water3$data)-1)) {
        for (rows in (columns+1):ncol(match_asv_phylo_water3$data)) {
                
                rand.vals = rand_w_bmntd_comp[rows,columns,];
                weighted_bnti_water3[rows,columns] = (bmntd_water3[rows,columns] - mean(rand.vals)) / sd(rand.vals);
                rm("rand.vals");
                
        };
};

rownames(weighted_bnti_water3) = colnames(match_asv_phylo_water3$data)
colnames(weighted_bnti_water3) = colnames(match_asv_phylo_water3$data)

weighted_bnti_water3
hist(weighted_bnti_water3)
abline(v = c(-2,2), col = "red", lwd = c(2,2), lty = c(3,3))

##water 0.2 ?m
bmntd_w_water02 <- as.matrix(comdistnt(t(match_asv_phylo_water02$data), cophenetic(match_asv_phylo_water02$phy), 
                                    abundance.weighted = T))
dim(bmntd_w_water02)
bmntd_w_water02[1:5, 1:5]

#make sure names match, should both be true
identical(colnames(match_asv_phylo_water02$data), colnames(bmntd_w_water02))
identical(colnames(match_asv_phylo_water02$data), rownames(bmntd_w_water02))


#calculate randomized betaMNTD (tip randomization)
beta.reps = 999

rand_w_bmntd_comp <- array(c(-999),dim=c(ncol(match_asv_phylo_water02$data),
                                         ncol(match_asv_phylo_water02$data),
                                         beta.reps))
dim(rand_w_bmntd_comp)

for (rep in 1:beta.reps) {
        
        rand_w_bmntd_comp[,,rep] = as.matrix(comdistnt(t(match_asv_phylo_water02$data),
                                                       taxaShuffle(cophenetic(match_asv_phylo_water02$phy)),
                                                       abundance.weighted=T,
                                                       exclude.conspecifics = F));
        
        print(c(date(),rep));
        
}


#weighted betaNTI
weighted_bnti_water02 <- matrix(c(NA), nrow = ncol(match_asv_phylo_water02$data),
                             ncol = ncol(match_asv_phylo_water02$data))
dim(weighted_bnti_water02)


for (columns in 1:(ncol(match_asv_phylo_water02$data)-1)) {
        for (rows in (columns+1):ncol(match_asv_phylo_water02$data)) {
                
                rand.vals = rand_w_bmntd_comp[rows,columns,];
                weighted_bnti_water02[rows,columns] = (bmntd_w_water02[rows,columns] - mean(rand.vals)) / sd(rand.vals);
                rm("rand.vals");
                
        };
};

rownames(weighted_bnti_water02) = colnames(match_asv_phylo_water02$data)
colnames(weighted_bnti_water02) = colnames(match_asv_phylo_water02$data)

weighted_bnti_water02
hist(weighted_bnti_water02)
abline(v = c(-2,2), col = "red", lwd = c(2,2), lty = c(3,3))


# Raup-Crick metric based on Bray-Curtis dissimilarities
raup_crick_abu_par <- function(com, reps, ncore, classic_metric=FALSE, split_ties=TRUE){
  
  
  require("parallel")
  require("doSNOW")
  
  pb <- txtProgressBar(max =reps, style = 3)
  progress <- function(n) setTxtProgressBar(pb, n)
  opts <- list(progress = progress)
  cl <- makeCluster(ncore)
  registerDoSNOW(cl)
  
  bray.rand <- foreach(randomize = 1:reps, 
                       .options.snow = opts,
                       .packages = c("vegan", "picante")) %dopar% {
                         
                         
                         null.dist <- com*0
                         
                         for(i in 1:nrow(com)){
                           
                           com.pa <- (com>0)*1
                           gamma<-ncol(com)
                           occur<-apply(com>0, MARGIN=2, FUN=sum)
                           abundance<-apply(com, MARGIN=2, FUN=sum)
                           com1 <- rep(0,gamma)
                           
                           com1[sample(1:gamma, sum(com.pa[i,]), replace=FALSE, prob=occur)]<-1
                           com1.samp.sp = sample(which(com1>0), (sum(com[i,])-sum(com1)),
                                                 replace=TRUE,prob=abundance[which(com1>0)]);
                           com1.samp.sp = cbind(com1.samp.sp,1)
                           com1.sp.counts = as.data.frame(tapply(com1.samp.sp[,2],com1.samp.sp[,1],FUN=sum))
                           colnames(com1.sp.counts) = 'counts'
                           com1.sp.counts$sp = as.numeric(rownames(com1.sp.counts))
                           com1[com1.sp.counts$sp] = com1[com1.sp.counts$sp] + com1.sp.counts$counts
                           x <- com1
                           null.dist[i,] <- x
                           rm('com1.samp.sp','com1.sp.counts')
                         }
                         as.matrix(vegdist(null.dist, "bray"))
                       }
  stopCluster(cl)
  
  ## Calculate beta-diversity for obs metacommunity
  bray.obs <- as.matrix(vegdist(com, "bray"))
  
  ##how many null observations is the observed value tied with?
  null_bray_curtis <- bray.rand
  num_exact_matching_in_null <- lapply(null_bray_curtis, function(x) x==bray.obs)
  num_exact_matching_in_null <- apply(simplify2array(num_exact_matching_in_null), 1:2, sum)
  
  ##how many null values are smaller than the observed *dissimilarity*?
  num_less_than_in_null <- lapply(null_bray_curtis, function(x) (x<bray.obs)*1)
  num_less_than_in_null <- apply(simplify2array(num_less_than_in_null), 1:2, sum)
  
  
  rc = (num_less_than_in_null)/reps; # rc;
  
  if(split_ties){
    
    rc = ((num_less_than_in_null +(num_exact_matching_in_null)/2)/reps)
  };
  
  
  if(!classic_metric){
    
    ##our modification of raup crick standardizes the metric to range from -1 to 1 instead of 0 to 1
    
    rc = (rc-.5)*2
  };
  
  return(rc)
  
}

#by type
RCbray_hdpe <- raup_crick_abu_par(sp_hdpe, 999, 16)
RCbray_hdpe

RCbray_tw <- raup_crick_abu_par(sp_tw, 999, 16)
RCbray_tw

RCbray_wood <- raup_crick_abu_par(sp_wood, 999, 16)
RCbray_wood

RCbray_water_3 <- raup_crick_abu_par(sp_water_3, 999, 16)
RCbray_water_3

RCbray_water_0_2 <- raup_crick_abu_par(sp_water_0_2, 999, 16)
RCbray_water_0_2
```

### Visualization
```{r plots}
# Diversity plots
## Alpha diversity (box plots)
box_theme <- theme(axis.ticks.y = element_line(size = 0.75),
         axis.ticks.length.y = unit(0.2, "cm"),
         axis.text.y = element_text(colour = "black", size = 16),
         axis.ticks.x = element_blank(),
         axis.text.x = element_blank(),
         strip.background = element_blank(),
         panel.background = element_rect(fill = "white", colour = "#000000", size = 0.7),
         panel.border = element_rect(fill = NA, colour = "black", size = 0.75),
         panel.grid.major = element_blank(),
         panel.grid.minor = element_blank(),
         legend.text = element_text(colour = "black", size = 16),
         legend.background = element_blank(),
         legend.key = element_blank(),
         legend.key.size = unit(0.9, "cm"),
         legend.position = "bottom",
         legend.direction = "horizontal",
         legend.title = element_text(colour = "black", face = "bold", size = 16),
         plot.margin = unit(c(1,1,1,1), "lines"))

label_type <- c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")
names(label_type) <- c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2")


(shannon_plot <- ggplot(data_scores_alpha, aes(x = site_x_cage, y = shannon, colour = site_x_cage, fill = site_x_cage)) +
    geom_boxplot(position = position_dodge(1), alpha = 0.7, lwd = 0.6) +
    theme(axis.title.y = element_text(colour = "black", size = 16, vjust = 3),
         axis.title.x = element_blank(),
         strip.text = element_text(color = "black", size = 16)) +
    scale_x_discrete(name = "Sampling Site", labels = element_blank(), 
                    limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s")) +
    scale_y_continuous(name = "Shannon") +
    labs(colour = "Sampling Site", fill = "Sampling Site") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                       limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"),  
                       labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                     limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                     labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    facet_wrap(~type, ncol = 5, labeller = labeller(type = label_type)) +
    box_theme)

(simpson_plot <- ggplot(data_scores_alpha, aes(x = site_x_cage, y = simpsons, colour = site_x_cage, fill = site_x_cage)) +
    geom_boxplot(position = position_dodge(1), alpha = 0.7, lwd = 0.6) +
    theme(axis.title.y = element_text(colour = "black", size = 16, vjust = 3),
         axis.title.x = element_blank(),
         strip.text = element_blank()) +
    scale_x_discrete(name = "Sampling Site", labels = element_blank(), 
                     limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s")) +
    scale_y_continuous(name = "Gini-Simpson") +
    labs(colour = "Sampling Site", fill = "Sampling Site") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                        limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"),  
                        labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                      limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                      labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    facet_wrap(~type, ncol = 5, labeller = labeller(type = label_type)) +
    box_theme)

(faith_plot <- ggplot(data_scores_phy_alpha, aes(x = site_x_cage, y = PD, colour = site_x_cage, fill = site_x_cage)) +
    geom_boxplot(position = position_dodge(1), alpha = 0.7, lwd = 0.6) +
    theme(axis.title.y = element_text(colour = "black", size = 16, vjust = 3),
         axis.title.x = element_blank(),
         strip.text = element_blank()) +
    scale_x_discrete(name = "Sampling Site", labels = element_blank(), 
                     limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s")) +
    scale_y_continuous(name = "Faith's PD") +
    labs(colour = "Sampling Site", fill = "Sampling Site") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                        limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"),  
                        labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                      limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                      labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    facet_wrap(~type, ncol = 5, labeller = labeller(type = label_type)) +
    box_theme)

(pd_ses_plot <- ggplot(data_scores_phy_alpha, aes(x = site_x_cage, y = PD.ses, colour = site_x_cage, fill = site_x_cage)) +
    geom_boxplot(position = position_dodge(1), alpha = 0.7, lwd = 0.6) +
    theme(axis.title.y = element_text(colour = "black", size = 16, vjust = 3),
         axis.title.x = element_blank(),
         strip.text = element_blank()) +
    scale_x_discrete(name = "Sampling Site", labels = element_blank(), 
                     limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s")) +
    scale_y_continuous(name = "Faith's PD (SES)") +
    labs(colour = "Sampling Site", fill = "Sampling Site") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                        limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"),  
                        llabels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                      limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                      labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    facet_wrap(~type, ncol = 5, labeller = labeller(type = label_type)) +
    box_theme)


(nti_plot <- ggplot(data_scores_phy_alpha, aes(x = site_x_cage, y = NTI_uw, colour = site_x_cage, fill = site_x_cage)) +
    geom_boxplot(position = position_dodge(1), alpha = 0.7, lwd = 0.6) +
    theme(axis.title.y = element_text(colour = "black", size = 16, vjust = 3),
         axis.title.x = element_blank(),
         strip.text = element_blank()) +
    scale_x_discrete(name = "Sampling Site", labels = element_blank(), 
                     limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s")) +
    scale_y_continuous(name = "NTI") +
    labs(colour = "Sampling Site", fill = "Sampling Site") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                        limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"),  
                        labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                      limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                      labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    facet_wrap(~type, ncol = 5, labeller = labeller(type = label_type)) +
    box_theme)

# combine plots
(alpha_plot <- ggarrange(shannon_plot, simpson_plot, faith_plot, pd_ses_plot, nti_plot, 
                         common.legend = T, legend = "bottom", ncol = 1, nrow = 5, align = "hv"))

ggsave("alpha_site_plot.tiff", alpha_site, dpi = 300, width = 250, height = 350, units = "mm")


## Beta diversity (PCoA plots)
beta_theme <- theme(axis.text.y = element_text(colour = "black", size = 14),
                    axis.text.x = element_text(colour = "black", size = 14),
                    axis.ticks = element_line(size = 0.75),
                    axis.ticks.length = unit(0.2, "cm"),
                    axis.title.y = element_text(colour = "black", size = 14, vjust = 2),
                    axis.title.x = element_text(colour = "black", size = 14, vjust = -2),
                    legend.title = element_text(colour = "black", face = "bold", size = 16),
                    legend.position = "right",
                    legend.key = element_blank(),
                    legend.text = element_text(colour = "black", size = 16),
                    panel.background = element_blank(), 
                    panel.border = element_rect(fill = NA, colour = "black", size = 0.75),
                    plot.margin = margin(t = 0, r = 1, b = 1, l = 1, unit = "cm"))
                    

(bc_pcoa_bremen <- ggplot(PCoA_BC_bremen_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) + 
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_BC_bremen$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_BC_bremen$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    ggtitle("Bray Curtis") +
    beta_theme)

(jc_pcoa_bremen <- ggplot(PCoA_JC_bremen_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) + 
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_JC_bremen$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_JC_bremen$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    ggtitle("Jaccard") +
    beta_theme)


(wUF_pcoa_bremen <- ggplot(PCoA_wUniF_vectors_bremen, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) + 
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_wUniF_bremen$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_wUniF_bremen$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    ggtitle("Weighted UniFrac") +
    beta_theme)


(uUF_pcoa_bremen <- ggplot(PCoA_uUniF_vectors_bremen, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) + 
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_uUniF_bremen$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_uUniF_bremen$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    ggtitle("Unweighted UniFrac") +
    beta_theme)

(bc_pcoa_brake <- ggplot(PCoA_BC_brake_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) + 
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_BC_brake$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_BC_brake$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme)

(jc_pcoa_brake <- ggplot(PCoA_JC_brake_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_JC_brake$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_JC_brake$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme) 


(wUF_pcoa_brake <- ggplot(PCoA_wUniF_vectors_brake, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_wUniF_brake$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_wUniF_brake$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme) 


(uUF_pcoa_brake <- ggplot(PCoA_uUniF_vectors_brake, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_uUniF_brake$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_uUniF_brake$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme) 


(bc_pcoa_bremerhaven <- ggplot(PCoA_BC_bremerhaven_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_BC_bremerhaven$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_BC_bremerhaven$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme)

(jc_pcoa_bremerhaven <- ggplot(PCoA_JC_bremerhaven_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_JC_bremerhaven$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_JC_bremerhaven$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme) 


(wUF_pcoa_bremerhaven <- ggplot(PCoA_wUniF_vectors_bremerhaven, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_wUniF_bremerhaven$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_wUniF_bremerhaven$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme) 


(uUF_pcoa_bremerhaven <- ggplot(PCoA_uUniF_vectors_bremerhaven, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_uUniF_bremerhaven$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_uUniF_bremerhaven$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme) 


(bc_pcoa_helgoland <- ggplot(PCoA_BC_helgoland_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_BC_helgoland$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_BC_helgoland$values$Relative_eig[2] * 100, 1), "%]"), 
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme)

(jc_pcoa_helgoland <- ggplot(PCoA_JC_helgoland_vectors, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_JC_helgoland$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_JC_helgoland$values$Relative_eig[2] * 100, 1), "%]"),
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme)

(wUF_pcoa_helgoland <- ggplot(PCoA_wUniF_vectors_helgoland, aes(x = Axis.1, y = Axis.2)) + 
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_wUniF_helgoland$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_wUniF_helgoland$values$Relative_eig[2] * 100, 1), "%]"),
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme)


(uUF_pcoa_helgoland <- ggplot(PCoA_uUniF_vectors_helgoland, aes(x = Axis.1, y = Axis.2)) +
    geom_point(size = 4, aes(shape = type, fill = site_x_cage, colour = site_x_cage), alpha = 0.8, stroke = 1) +
    theme(plot.title = element_text(hjust = 0, face = "bold", colour = "black", size = 16)) +
    guides(shape = guide_legend(order = 2), fill = guide_legend(order = 1), colour = guide_legend(order = 1)) +
    geom_hline(aes(yintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) + 
    geom_vline(aes(xintercept = 0), colour = c("#666666"), linetype = "dotted", size = 0.5) +
    labs(x = paste0("Axis 1 [", round(PCoA_uUniF_helgoland$values$Relative_eig[1] * 100, 1), "%]"),
         y = paste0("Axis 2 [", round(PCoA_uUniF_helgoland$values$Relative_eig[2] * 100, 1), "%]"),
         colour = "Sampling Site", fill = "Sampling Site", shape = "Sample Type") +
    scale_colour_manual(values = c("#7E0000", "#CB2300", "#C8B332", "#0082BD", "#000B7D"), 
                            limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                            labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_fill_manual(values = c("#A50026", "#F46D43", "#fae48b", "#74ADD1", "#313695"), 
                          limits = c("a_bremen", "b_brake", "c_bremerhaven", "d_helgoland", "e_helgoland_s"), 
                          labels = c("Bremen", "Brake", "Bremerhaven", "Helgoland (T)", "Helgoland (S)")) +
    scale_shape_manual(values = c(21, 22, 23, 24, 25),
                           limits = c("HDPE", "tyre_wear", "wood", "water_3", "water_0_2"), 
                           labels = c("HDPE", "Tyre Wear", "Wood", "Water 3", "Water 02")) +
    beta_theme)


# combine PCoA plots
(beta_site <- ggarrange(bc_pcoa_bremen, jc_pcoa_bremen, wUF_pcoa_bremen, uUF_pcoa_bremen,
                        bc_pcoa_brake, jc_pcoa_brake, wUF_pcoa_brake, uUF_pcoa_brake,
                        bc_pcoa_bremerhaven, jc_pcoa_bremerhaven, wUF_pcoa_bremerhaven, uUF_pcoa_bremerhaven,
                        bc_pcoa_helgoland, jc_pcoa_helgoland, wUF_pcoa_helgoland, uUF_pcoa_helgoland,
                        labels = NA, common.legend = T, legend = "right", ncol = 4, nrow = 4, align = "hv"))


ggsave("beta_site_plot.tiff", beta_site, dpi = 300, height = 297, width = 420, units = "mm")


```
